--- joern
+++ before
@@ -6,9 +6,6 @@
   METHOD_PARAMETER_IN NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
   METHOD_PARAMETER_OUT NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
   BLOCK CODE={\n#if 0\n    return first(x);\n#else\n    return second(x);\n#endif\n} TYPE_FULL_NAME=void ORDER=2
-    RETURN CODE=return second(x); ORDER=1
-      CALL NAME=second CODE=second(x) TYPE_FULL_NAME=int METHOD_FULL_NAME=second ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
-        IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
 
 METHOD NAME=first CODE=int first(int x); FULL_NAME=first SIGNATURE=int(int) ORDER=1
@@ -22,9 +19,6 @@
     METHOD_PARAMETER_IN NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
     METHOD_PARAMETER_OUT NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
     BLOCK CODE={\n#if 0\n    return first(x);\n#else\n    return second(x);\n#endif\n} TYPE_FULL_NAME=void ORDER=2
-      RETURN CODE=return second(x); ORDER=1
-        CALL NAME=second CODE=second(x) TYPE_FULL_NAME=int METHOD_FULL_NAME=second ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
-          IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
     METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
   BLOCK TYPE_FULL_NAME=ANY ORDER=1
     METHOD_REF CODE=choose TYPE_FULL_NAME=choose METHOD_FULL_NAME=choose ORDER=1
@@ -53,17 +47,11 @@
 NODES|TYPE NAME=choose FULL_NAME=choose TYPE_DECL_FULL_NAME=choose
 NODES|TYPE NAME=int FULL_NAME=int TYPE_DECL_FULL_NAME=int
 NODES|TYPE NAME=void FULL_NAME=void TYPE_DECL_FULL_NAME=void
-EDGES|ARGUMENT choose#4 -> choose#5
-EDGES|ARGUMENT choose#5 -> choose#6
-EDGES|CALL choose#5 -> second#0
 EDGES|CFG <includes>:<global>#0 -> <includes>:<global>#2
-EDGES|CFG choose#0 -> choose#6
-EDGES|CFG choose#4 -> choose#7
-EDGES|CFG choose#5 -> choose#4
-EDGES|CFG choose#6 -> choose#5
+EDGES|CFG choose#0 -> choose#4
 EDGES|CFG first#0 -> first#4
-EDGES|CFG main.c:<global>#0 -> main.c:<global>#10
-EDGES|CFG main.c:<global>#10 -> main.c:<global>#11
+EDGES|CFG main.c:<global>#0 -> main.c:<global>#7
+EDGES|CFG main.c:<global>#7 -> main.c:<global>#8
 EDGES|CFG second#0 -> second#4
 EDGES|CONTAINS <includes>:<global>#0 -> <includes>:<global>#1
 EDGES|CONTAINS D:main.c:<global> -> D:choose
@@ -76,29 +64,24 @@
 EDGES|CONTAINS F:<includes> -> D:void
 EDGES|CONTAINS F:main.c -> D:main.c:<global>
 EDGES|CONTAINS choose#0 -> choose#3
-EDGES|CONTAINS choose#0 -> choose#4
-EDGES|CONTAINS choose#0 -> choose#5
-EDGES|CONTAINS choose#0 -> choose#6
 EDGES|CONTAINS first#0 -> first#3
 EDGES|CONTAINS main.c:<global>#0 -> choose#0
-EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#10
-EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#9
+EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#6
+EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#7
 EDGES|CONTAINS second#0 -> second#3
 EDGES|EVAL_TYPE <includes>:<global>#1 -> T:ANY
 EDGES|EVAL_TYPE <includes>:<global>#2 -> T:ANY
 EDGES|EVAL_TYPE choose#1 -> T:int
 EDGES|EVAL_TYPE choose#2 -> T:int
 EDGES|EVAL_TYPE choose#3 -> T:void
-EDGES|EVAL_TYPE choose#5 -> T:int
-EDGES|EVAL_TYPE choose#6 -> T:int
-EDGES|EVAL_TYPE choose#7 -> T:int
+EDGES|EVAL_TYPE choose#4 -> T:int
 EDGES|EVAL_TYPE first#1 -> T:int
 EDGES|EVAL_TYPE first#2 -> T:int
 EDGES|EVAL_TYPE first#3 -> T:ANY
 EDGES|EVAL_TYPE first#4 -> T:int
-EDGES|EVAL_TYPE main.c:<global>#10 -> T:choose
-EDGES|EVAL_TYPE main.c:<global>#11 -> T:ANY
-EDGES|EVAL_TYPE main.c:<global>#9 -> T:ANY
+EDGES|EVAL_TYPE main.c:<global>#6 -> T:ANY
+EDGES|EVAL_TYPE main.c:<global>#7 -> T:choose
+EDGES|EVAL_TYPE main.c:<global>#8 -> T:ANY
 EDGES|EVAL_TYPE second#1 -> T:int
 EDGES|EVAL_TYPE second#2 -> T:int
 EDGES|EVAL_TYPE second#3 -> T:ANY
@@ -113,8 +96,7 @@
 EDGES|REF T:choose -> D:choose
 EDGES|REF T:int -> D:int
 EDGES|REF T:void -> D:void
-EDGES|REF choose#6 -> choose#1
-EDGES|REF main.c:<global>#10 -> choose#0
+EDGES|REF main.c:<global>#7 -> choose#0
 EDGES|SOURCE_FILE <includes>:<global>#0 -> F:<includes>
 EDGES|SOURCE_FILE D:ANY -> F:<includes>
 EDGES|SOURCE_FILE D:choose -> F:main.c
@@ -128,19 +110,12 @@
 EDGES|SOURCE_FILE first#0 -> F:main.c
 EDGES|SOURCE_FILE main.c:<global>#0 -> F:main.c
 EDGES|SOURCE_FILE second#0 -> F:main.c
-FLOWS|REACHING_DEF[<RET>] choose#4 -> choose#7
 FLOWS|REACHING_DEF[] choose#0 -> choose#1
-FLOWS|REACHING_DEF[] choose#0 -> choose#6
 FLOWS|REACHING_DEF[] first#0 -> first#1
-FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#10
+FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#7
 FLOWS|REACHING_DEF[] second#0 -> second#1
-FLOWS|REACHING_DEF[second(x)] choose#5 -> choose#4
-FLOWS|REACHING_DEF[second(x)] choose#5 -> choose#7
 FLOWS|REACHING_DEF[x] choose#1 -> choose#2
-FLOWS|REACHING_DEF[x] choose#1 -> choose#6
-FLOWS|REACHING_DEF[x] choose#6 -> choose#2
-FLOWS|REACHING_DEF[x] choose#6 -> choose#5
-FLOWS|REACHING_DEF[x] choose#6 -> choose#7
+FLOWS|REACHING_DEF[x] choose#1 -> choose#4
 FLOWS|REACHING_DEF[x] first#1 -> first#2
 FLOWS|REACHING_DEF[x] first#1 -> first#4
 FLOWS|REACHING_DEF[x] second#1 -> second#2
