--- live
+++ baseline
@@ -4,8 +4,7 @@
 
 METHOD NAME=<global> CODE=<global> FULL_NAME=registration.c:<global> ORDER=1
   BLOCK TYPE_FULL_NAME=ANY ORDER=1
-    TYPE_DECL NAME=Packet CODE=struct Packet FULL_NAME=Packet ORDER=1
-    LOCAL NAME=packet CODE=struct Packet *packet TYPE_FULL_NAME=Packet* ORDER=2
+    LOCAL NAME=packet CODE=struct Packet *packet TYPE_FULL_NAME=Packet* ORDER=1
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
 
 NODES|META_DATA LANGUAGE=NEWC
@@ -17,44 +16,39 @@
 NODES|NAMESPACE_BLOCK NAME=<global> FULL_NAME=registration.c:<global> FILENAME=registration.c ORDER=1
 NODES|NAMESPACE NAME=<global>
 NODES|TYPE_DECL NAME=ANY FULL_NAME=ANY CODE=ANY IS_EXTERNAL=true AST_PARENT_TYPE=NAMESPACE_BLOCK AST_PARENT_FULL_NAME=<includes>:<global> FILENAME=<includes>
-NODES|TYPE_DECL NAME=Packet FULL_NAME=Packet CODE=struct Packet AST_PARENT_TYPE= AST_PARENT_FULL_NAME= FILENAME=registration.c ORDER=1
+NODES|TYPE_DECL NAME=Packet FULL_NAME=Packet CODE=Packet IS_EXTERNAL=true AST_PARENT_TYPE=NAMESPACE_BLOCK AST_PARENT_FULL_NAME=<includes>:<global> FILENAME=<includes>
 NODES|TYPE_DECL NAME=Packet* FULL_NAME=Packet* CODE=Packet* IS_EXTERNAL=true AST_PARENT_TYPE=NAMESPACE_BLOCK AST_PARENT_FULL_NAME=<includes>:<global> FILENAME=<includes>
-NODES|TYPE_DECL NAME=packet FULL_NAME=packet CODE=packet IS_EXTERNAL=true AST_PARENT_TYPE=NAMESPACE_BLOCK AST_PARENT_FULL_NAME=<includes>:<global> FILENAME=<includes>
 NODES|TYPE_DECL NAME=<global> FULL_NAME=registration.c:<global> CODE=<global> AST_PARENT_TYPE=NAMESPACE_BLOCK AST_PARENT_FULL_NAME=registration.c:<global> FILENAME=registration.c ORDER=1
 NODES|TYPE NAME=ANY FULL_NAME=ANY TYPE_DECL_FULL_NAME=ANY
 NODES|TYPE NAME=Packet FULL_NAME=Packet TYPE_DECL_FULL_NAME=Packet
 NODES|TYPE NAME=Packet* FULL_NAME=Packet* TYPE_DECL_FULL_NAME=Packet*
-NODES|TYPE NAME=packet FULL_NAME=packet TYPE_DECL_FULL_NAME=packet
 EDGES|CFG <includes>:<global>#0 -> <includes>:<global>#2
-EDGES|CFG registration.c:<global>#0 -> registration.c:<global>#4
+EDGES|CFG registration.c:<global>#0 -> registration.c:<global>#3
 EDGES|CONTAINS <includes>:<global>#0 -> <includes>:<global>#1
 EDGES|CONTAINS D:registration.c:<global> -> registration.c:<global>#0
 EDGES|CONTAINS F:<includes> -> <includes>:<global>#0
 EDGES|CONTAINS F:<includes> -> D:ANY
+EDGES|CONTAINS F:<includes> -> D:Packet
 EDGES|CONTAINS F:<includes> -> D:Packet*
-EDGES|CONTAINS F:<includes> -> D:packet
 EDGES|CONTAINS F:registration.c -> D:registration.c:<global>
 EDGES|CONTAINS registration.c:<global>#0 -> registration.c:<global>#1
-EDGES|CONTAINS registration.c:<global>#0 -> registration.c:<global>#2
 EDGES|EVAL_TYPE <includes>:<global>#1 -> T:ANY
 EDGES|EVAL_TYPE <includes>:<global>#2 -> T:ANY
 EDGES|EVAL_TYPE registration.c:<global>#1 -> T:ANY
-EDGES|EVAL_TYPE registration.c:<global>#3 -> T:Packet*
-EDGES|EVAL_TYPE registration.c:<global>#4 -> T:ANY
+EDGES|EVAL_TYPE registration.c:<global>#2 -> T:Packet*
+EDGES|EVAL_TYPE registration.c:<global>#3 -> T:ANY
 EDGES|REF NB:<global> -> NS:<global>
 EDGES|REF NB:<includes>:<global> -> NS:<global>
 EDGES|REF NB:registration.c:<global> -> NS:<global>
 EDGES|REF T:ANY -> D:ANY
-EDGES|REF T:Packet -> registration.c:<global>#2
+EDGES|REF T:Packet -> D:Packet
 EDGES|REF T:Packet* -> D:Packet*
-EDGES|REF T:packet -> D:packet
 EDGES|SOURCE_FILE <includes>:<global>#0 -> F:<includes>
 EDGES|SOURCE_FILE D:ANY -> F:<includes>
+EDGES|SOURCE_FILE D:Packet -> F:<includes>
 EDGES|SOURCE_FILE D:Packet* -> F:<includes>
-EDGES|SOURCE_FILE D:packet -> F:<includes>
 EDGES|SOURCE_FILE D:registration.c:<global> -> F:registration.c
 EDGES|SOURCE_FILE NB:<global> -> F:<unknown>
 EDGES|SOURCE_FILE NB:<includes>:<global> -> F:<includes>
 EDGES|SOURCE_FILE NB:registration.c:<global> -> F:registration.c
 EDGES|SOURCE_FILE registration.c:<global>#0 -> F:registration.c
-EDGES|SOURCE_FILE registration.c:<global>#2 -> F:registration.c
