--- joern
+++ final-after-v3
@@ -51,7 +51,6 @@
 NODES|TYPE_DECL NAME=g FULL_NAME=g CODE=g AST_PARENT_TYPE=TYPE_DECL AST_PARENT_FULL_NAME=main.c:<global> FILENAME=main.c ORDER=1
 NODES|TYPE_DECL NAME=int FULL_NAME=int CODE=int IS_EXTERNAL=true AST_PARENT_TYPE=NAMESPACE_BLOCK AST_PARENT_FULL_NAME=<includes>:<global> FILENAME=<includes>
 NODES|TYPE_DECL NAME=long FULL_NAME=long CODE=long IS_EXTERNAL=true AST_PARENT_TYPE=NAMESPACE_BLOCK AST_PARENT_FULL_NAME=<includes>:<global> FILENAME=<includes>
-NODES|TYPE_DECL NAME=longint FULL_NAME=longint CODE=longint IS_EXTERNAL=true AST_PARENT_TYPE=NAMESPACE_BLOCK AST_PARENT_FULL_NAME=<includes>:<global> FILENAME=<includes>
 NODES|TYPE_DECL NAME=<global> FULL_NAME=main.c:<global> CODE=<global> AST_PARENT_TYPE=NAMESPACE_BLOCK AST_PARENT_FULL_NAME=main.c:<global> FILENAME=main.c ORDER=2
 NODES|TYPE_DECL NAME=void FULL_NAME=void CODE=void IS_EXTERNAL=true AST_PARENT_TYPE=NAMESPACE_BLOCK AST_PARENT_FULL_NAME=<includes>:<global> FILENAME=<includes>
 NODES|TYPE NAME=ANY FULL_NAME=ANY TYPE_DECL_FULL_NAME=ANY
@@ -59,7 +58,6 @@
 NODES|TYPE NAME=g FULL_NAME=g TYPE_DECL_FULL_NAME=g
 NODES|TYPE NAME=int FULL_NAME=int TYPE_DECL_FULL_NAME=int
 NODES|TYPE NAME=long FULL_NAME=long TYPE_DECL_FULL_NAME=long
-NODES|TYPE NAME=longint FULL_NAME=longint TYPE_DECL_FULL_NAME=longint
 NODES|TYPE NAME=void FULL_NAME=void TYPE_DECL_FULL_NAME=void
 EDGES|ARGUMENT g#4 -> g#5
 EDGES|ARGUMENT g#5 -> g#6
@@ -82,7 +80,6 @@
 EDGES|CONTAINS F:<includes> -> D:ANY
 EDGES|CONTAINS F:<includes> -> D:int
 EDGES|CONTAINS F:<includes> -> D:long
-EDGES|CONTAINS F:<includes> -> D:longint
 EDGES|CONTAINS F:<includes> -> D:void
 EDGES|CONTAINS F:api.h -> D:api.h:<global>
 EDGES|CONTAINS F:main.c -> D:main.c:<global>
@@ -124,7 +121,6 @@
 EDGES|REF T:g -> D:g
 EDGES|REF T:int -> D:int
 EDGES|REF T:long -> D:long
-EDGES|REF T:longint -> D:longint
 EDGES|REF T:void -> D:void
 EDGES|REF g#6 -> g#1
 EDGES|REF main.c:<global>#10 -> g#0
@@ -134,7 +130,6 @@
 EDGES|SOURCE_FILE D:g -> F:main.c
 EDGES|SOURCE_FILE D:int -> F:<includes>
 EDGES|SOURCE_FILE D:long -> F:<includes>
-EDGES|SOURCE_FILE D:longint -> F:<includes>
 EDGES|SOURCE_FILE D:main.c:<global> -> F:main.c
 EDGES|SOURCE_FILE D:void -> F:<includes>
 EDGES|SOURCE_FILE NB:<global> -> F:<unknown>
