--- Joern
+++ before
@@ -1,6 +1,22 @@
 METHOD NAME=<global> CODE=<global> FULL_NAME=<includes>:<global> ORDER=1
   BLOCK TYPE_FULL_NAME=ANY ORDER=1
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
+
+METHOD NAME=<operator>.assignment FULL_NAME=<operator>.assignment ORDER=0
+  METHOD_PARAMETER_IN NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
+  BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
+  METHOD_PARAMETER_OUT NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
+  METHOD_PARAMETER_IN NAME=p2 CODE=p2 TYPE_FULL_NAME=ANY ORDER=2
+  METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
+  METHOD_PARAMETER_OUT NAME=p2 CODE=p2 TYPE_FULL_NAME=ANY ORDER=2
+
+METHOD NAME=<operator>.notEquals FULL_NAME=<operator>.notEquals ORDER=0
+  METHOD_PARAMETER_IN NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
+  BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
+  METHOD_PARAMETER_OUT NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
+  METHOD_PARAMETER_IN NAME=p2 CODE=p2 TYPE_FULL_NAME=ANY ORDER=2
+  METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
+  METHOD_PARAMETER_OUT NAME=p2 CODE=p2 TYPE_FULL_NAME=ANY ORDER=2
 
 METHOD NAME=<operator>.postDecrement FULL_NAME=<operator>.postDecrement ORDER=0
   METHOD_PARAMETER_IN NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
@@ -19,12 +35,31 @@
     METHOD_PARAMETER_IN NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
     METHOD_PARAMETER_OUT NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
     BLOCK CODE={ for(int y=out(x,F,M(x));y;x--) { x=y; } return M(x); } TYPE_FULL_NAME=void ORDER=2
-      LOCAL NAME=y CODE=<unknown> y TYPE_FULL_NAME=ANY ORDER=0
-      UNKNOWN CODE=for(int y=out(x,F,M(x)); ORDER=1
-      IDENTIFIER NAME=y CODE=<unknown> y TYPE_FULL_NAME=ANY ORDER=2
-      CALL NAME=<operator>.postDecrement CODE=x-- TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.postDecrement ORDER=3 DISPATCH_TYPE=STATIC_DISPATCH
-        IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
-      RETURN CODE=return M(x); ORDER=4
+      LOCAL NAME=UNDEF CODE=<unknown> UNDEF TYPE_FULL_NAME=ANY ORDER=0
+      CONTROL_STRUCTURE CODE=for (int y=out(x,F,M(x));y;x--) ORDER=1
+        LOCAL NAME=y CODE=int y TYPE_FULL_NAME=int ORDER=1
+        CALL NAME=<operator>.assignment CODE=y=out(x,F,M(x)) TYPE_FULL_NAME=void METHOD_FULL_NAME=<operator>.assignment ORDER=2 ARGUMENT_INDEX=1 DISPATCH_TYPE=STATIC_DISPATCH
+          IDENTIFIER NAME=y CODE=y TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+          CALL NAME=out CODE=out(x,F,M(x)) TYPE_FULL_NAME=int METHOD_FULL_NAME=out ORDER=2 ARGUMENT_INDEX=2 DISPATCH_TYPE=STATIC_DISPATCH
+            IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+            CALL NAME=F CODE=F TYPE_FULL_NAME=char* METHOD_FULL_NAME=main.c:F:char*(0) SIGNATURE=char*(0) ORDER=2 ARGUMENT_INDEX=2 DISPATCH_TYPE=INLINED
+              BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
+                LITERAL CODE="%" UNDEF "d" TYPE_FULL_NAME=char* ORDER=1
+            CALL NAME=M CODE=M(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:M:ANY(1) SIGNATURE=ANY(1) ORDER=3 ARGUMENT_INDEX=3 DISPATCH_TYPE=INLINED
+              IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+              BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
+                CALL NAME=helper CODE=helper(x) TYPE_FULL_NAME=int METHOD_FULL_NAME=helper ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
+                  IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+        CALL NAME=<operator>.notEquals CODE=y != 0 TYPE_FULL_NAME=int METHOD_FULL_NAME=<operator>.notEquals ORDER=3 DISPATCH_TYPE=STATIC_DISPATCH
+          IDENTIFIER NAME=y CODE=y TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+          LITERAL CODE=0 TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
+        CALL NAME=<operator>.postDecrement CODE=x-- TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.postDecrement ORDER=4 DISPATCH_TYPE=STATIC_DISPATCH
+          IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+        BLOCK CODE={ x=y; } TYPE_FULL_NAME=void ORDER=5
+          CALL NAME=<operator>.assignment CODE=x=y TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.assignment ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
+            IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+            IDENTIFIER NAME=y CODE=y TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
+      RETURN CODE=return M(x); ORDER=2
         CALL NAME=M CODE=M(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:M:ANY(1) SIGNATURE=ANY(1) ORDER=1 DISPATCH_TYPE=INLINED
           IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
           BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
@@ -35,6 +70,10 @@
     METHOD_REF CODE=test TYPE_FULL_NAME=test METHOD_FULL_NAME=test ORDER=1
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
 
+METHOD NAME=F CODE=#define F "%" UNDEF "d" FULL_NAME=main.c:F:char*(0) SIGNATURE=char*(0) ORDER=1
+  BLOCK TYPE_FULL_NAME=ANY ORDER=1
+  METHOD_RETURN CODE=RET TYPE_FULL_NAME=char* ORDER=2
+
 METHOD NAME=M CODE=#define M(x) helper(x) FULL_NAME=main.c:M:ANY(1) SIGNATURE=ANY(1) ORDER=1
   METHOD_PARAMETER_IN NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
   METHOD_PARAMETER_OUT NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
@@ -55,12 +94,31 @@
   METHOD_PARAMETER_IN NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
   METHOD_PARAMETER_OUT NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
   BLOCK CODE={ for(int y=out(x,F,M(x));y;x--) { x=y; } return M(x); } TYPE_FULL_NAME=void ORDER=2
-    LOCAL NAME=y CODE=<unknown> y TYPE_FULL_NAME=ANY ORDER=0
-    UNKNOWN CODE=for(int y=out(x,F,M(x)); ORDER=1
-    IDENTIFIER NAME=y CODE=<unknown> y TYPE_FULL_NAME=ANY ORDER=2
-    CALL NAME=<operator>.postDecrement CODE=x-- TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.postDecrement ORDER=3 DISPATCH_TYPE=STATIC_DISPATCH
-      IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
-    RETURN CODE=return M(x); ORDER=4
+    LOCAL NAME=UNDEF CODE=<unknown> UNDEF TYPE_FULL_NAME=ANY ORDER=0
+    CONTROL_STRUCTURE CODE=for (int y=out(x,F,M(x));y;x--) ORDER=1
+      LOCAL NAME=y CODE=int y TYPE_FULL_NAME=int ORDER=1
+      CALL NAME=<operator>.assignment CODE=y=out(x,F,M(x)) TYPE_FULL_NAME=void METHOD_FULL_NAME=<operator>.assignment ORDER=2 ARGUMENT_INDEX=1 DISPATCH_TYPE=STATIC_DISPATCH
+        IDENTIFIER NAME=y CODE=y TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+        CALL NAME=out CODE=out(x,F,M(x)) TYPE_FULL_NAME=int METHOD_FULL_NAME=out ORDER=2 ARGUMENT_INDEX=2 DISPATCH_TYPE=STATIC_DISPATCH
+          IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+          CALL NAME=F CODE=F TYPE_FULL_NAME=char* METHOD_FULL_NAME=main.c:F:char*(0) SIGNATURE=char*(0) ORDER=2 ARGUMENT_INDEX=2 DISPATCH_TYPE=INLINED
+            BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
+              LITERAL CODE="%" UNDEF "d" TYPE_FULL_NAME=char* ORDER=1
+          CALL NAME=M CODE=M(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:M:ANY(1) SIGNATURE=ANY(1) ORDER=3 ARGUMENT_INDEX=3 DISPATCH_TYPE=INLINED
+            IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+            BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
+              CALL NAME=helper CODE=helper(x) TYPE_FULL_NAME=int METHOD_FULL_NAME=helper ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
+                IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+      CALL NAME=<operator>.notEquals CODE=y != 0 TYPE_FULL_NAME=int METHOD_FULL_NAME=<operator>.notEquals ORDER=3 DISPATCH_TYPE=STATIC_DISPATCH
+        IDENTIFIER NAME=y CODE=y TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+        LITERAL CODE=0 TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
+      CALL NAME=<operator>.postDecrement CODE=x-- TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.postDecrement ORDER=4 DISPATCH_TYPE=STATIC_DISPATCH
+        IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+      BLOCK CODE={ x=y; } TYPE_FULL_NAME=void ORDER=5
+        CALL NAME=<operator>.assignment CODE=x=y TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.assignment ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
+          IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+          IDENTIFIER NAME=y CODE=y TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
+    RETURN CODE=return M(x); ORDER=2
       CALL NAME=M CODE=M(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:M:ANY(1) SIGNATURE=ANY(1) ORDER=1 DISPATCH_TYPE=INLINED
         IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
         BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
@@ -87,36 +145,78 @@
 NODES|TYPE NAME=int FULL_NAME=int TYPE_DECL_FULL_NAME=int
 NODES|TYPE NAME=test FULL_NAME=test TYPE_DECL_FULL_NAME=test
 NODES|TYPE NAME=void FULL_NAME=void TYPE_DECL_FULL_NAME=void
-EDGES|ARGUMENT test#10 -> test#11
-EDGES|ARGUMENT test#13 -> test#14
+EDGES|ARGUMENT test#14 -> test#15
+EDGES|ARGUMENT test#17 -> test#18
+EDGES|ARGUMENT test#19 -> test#20
+EDGES|ARGUMENT test#19 -> test#21
+EDGES|ARGUMENT test#22 -> test#23
+EDGES|ARGUMENT test#25 -> test#26
+EDGES|ARGUMENT test#25 -> test#27
+EDGES|ARGUMENT test#28 -> test#29
+EDGES|ARGUMENT test#29 -> test#30
+EDGES|ARGUMENT test#32 -> test#33
 EDGES|ARGUMENT test#7 -> test#8
+EDGES|ARGUMENT test#7 -> test#9
 EDGES|ARGUMENT test#9 -> test#10
-EDGES|CALL test#10 -> main.c:M:ANY(1)#0
-EDGES|CALL test#13 -> helper#0
-EDGES|CALL test#7 -> <operator>.postDecrement#0
+EDGES|ARGUMENT test#9 -> test#11
+EDGES|ARGUMENT test#9 -> test#14
+EDGES|CALL test#11 -> main.c:F:char*(0)#0
+EDGES|CALL test#14 -> main.c:M:ANY(1)#0
+EDGES|CALL test#17 -> helper#0
+EDGES|CALL test#19 -> <operator>.notEquals#0
+EDGES|CALL test#22 -> <operator>.postDecrement#0
+EDGES|CALL test#25 -> <operator>.assignment#0
+EDGES|CALL test#29 -> main.c:M:ANY(1)#0
+EDGES|CALL test#32 -> helper#0
+EDGES|CALL test#7 -> <operator>.assignment#0
+EDGES|CALL test#9 -> out#0
 EDGES|CFG <includes>:<global>#0 -> <includes>:<global>#2
+EDGES|CFG <operator>.assignment#0 -> <operator>.assignment#5
+EDGES|CFG <operator>.notEquals#0 -> <operator>.notEquals#5
 EDGES|CFG <operator>.postDecrement#0 -> <operator>.postDecrement#4
 EDGES|CFG helper#0 -> helper#4
-EDGES|CFG main.c:<global>#0 -> main.c:<global>#18
-EDGES|CFG main.c:<global>#1 -> test#5
-EDGES|CFG main.c:<global>#18 -> main.c:<global>#19
+EDGES|CFG main.c:<global>#0 -> main.c:<global>#37
+EDGES|CFG main.c:<global>#1 -> test#8
+EDGES|CFG main.c:<global>#37 -> main.c:<global>#38
+EDGES|CFG main.c:F:char*(0)#0 -> main.c:F:char*(0)#2
 EDGES|CFG main.c:M:ANY(1)#0 -> main.c:M:ANY(1)#4
 EDGES|CFG out#0 -> out#8
-EDGES|CFG test#10 -> test#14
-EDGES|CFG test#10 -> test#9
-EDGES|CFG test#11 -> test#10
-EDGES|CFG test#13 -> test#9
-EDGES|CFG test#14 -> test#13
-EDGES|CFG test#5 -> test#6
-EDGES|CFG test#6 -> test#8
-EDGES|CFG test#7 -> test#11
-EDGES|CFG test#8 -> test#7
-EDGES|CFG test#9 -> test#15
+EDGES|CFG test#10 -> test#11
+EDGES|CFG test#11 -> test#13
+EDGES|CFG test#11 -> test#15
+EDGES|CFG test#13 -> test#15
+EDGES|CFG test#14 -> test#18
+EDGES|CFG test#14 -> test#9
+EDGES|CFG test#15 -> test#14
+EDGES|CFG test#17 -> test#9
+EDGES|CFG test#18 -> test#17
+EDGES|CFG test#19 -> test#26
+EDGES|CFG test#19 -> test#30
+EDGES|CFG test#20 -> test#21
+EDGES|CFG test#21 -> test#19
+EDGES|CFG test#22 -> test#20
+EDGES|CFG test#23 -> test#22
+EDGES|CFG test#25 -> test#23
+EDGES|CFG test#26 -> test#27
+EDGES|CFG test#27 -> test#25
+EDGES|CFG test#28 -> test#34
+EDGES|CFG test#29 -> test#28
+EDGES|CFG test#29 -> test#33
+EDGES|CFG test#30 -> test#29
+EDGES|CFG test#32 -> test#28
+EDGES|CFG test#33 -> test#32
+EDGES|CFG test#7 -> test#20
+EDGES|CFG test#8 -> test#10
+EDGES|CFG test#9 -> test#7
+EDGES|CONDITION test#5 -> test#19
 EDGES|CONTAINS <includes>:<global>#0 -> <includes>:<global>#1
+EDGES|CONTAINS <operator>.assignment#0 -> <operator>.assignment#2
+EDGES|CONTAINS <operator>.notEquals#0 -> <operator>.notEquals#2
 EDGES|CONTAINS <operator>.postDecrement#0 -> <operator>.postDecrement#2
 EDGES|CONTAINS D:main.c:<global> -> D:test
 EDGES|CONTAINS D:main.c:<global> -> helper#0
 EDGES|CONTAINS D:main.c:<global> -> main.c:<global>#0
+EDGES|CONTAINS D:main.c:<global> -> main.c:F:char*(0)#0
 EDGES|CONTAINS D:main.c:<global> -> main.c:M:ANY(1)#0
 EDGES|CONTAINS D:main.c:<global> -> out#0
 EDGES|CONTAINS F:<includes> -> <includes>:<global>#0
@@ -127,23 +227,54 @@
 EDGES|CONTAINS F:main.c -> D:main.c:<global>
 EDGES|CONTAINS helper#0 -> helper#3
 EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#1
-EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#17
-EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#18
+EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#36
+EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#37
 EDGES|CONTAINS main.c:<global>#1 -> test#10
 EDGES|CONTAINS main.c:<global>#1 -> test#11
 EDGES|CONTAINS main.c:<global>#1 -> test#12
 EDGES|CONTAINS main.c:<global>#1 -> test#13
 EDGES|CONTAINS main.c:<global>#1 -> test#14
+EDGES|CONTAINS main.c:<global>#1 -> test#15
+EDGES|CONTAINS main.c:<global>#1 -> test#16
+EDGES|CONTAINS main.c:<global>#1 -> test#17
+EDGES|CONTAINS main.c:<global>#1 -> test#18
+EDGES|CONTAINS main.c:<global>#1 -> test#19
+EDGES|CONTAINS main.c:<global>#1 -> test#20
+EDGES|CONTAINS main.c:<global>#1 -> test#21
+EDGES|CONTAINS main.c:<global>#1 -> test#22
+EDGES|CONTAINS main.c:<global>#1 -> test#23
+EDGES|CONTAINS main.c:<global>#1 -> test#24
+EDGES|CONTAINS main.c:<global>#1 -> test#25
+EDGES|CONTAINS main.c:<global>#1 -> test#26
+EDGES|CONTAINS main.c:<global>#1 -> test#27
+EDGES|CONTAINS main.c:<global>#1 -> test#28
+EDGES|CONTAINS main.c:<global>#1 -> test#29
 EDGES|CONTAINS main.c:<global>#1 -> test#3
+EDGES|CONTAINS main.c:<global>#1 -> test#30
+EDGES|CONTAINS main.c:<global>#1 -> test#31
+EDGES|CONTAINS main.c:<global>#1 -> test#32
+EDGES|CONTAINS main.c:<global>#1 -> test#33
 EDGES|CONTAINS main.c:<global>#1 -> test#5
-EDGES|CONTAINS main.c:<global>#1 -> test#6
 EDGES|CONTAINS main.c:<global>#1 -> test#7
 EDGES|CONTAINS main.c:<global>#1 -> test#8
 EDGES|CONTAINS main.c:<global>#1 -> test#9
+EDGES|CONTAINS main.c:F:char*(0)#0 -> main.c:F:char*(0)#1
 EDGES|CONTAINS main.c:M:ANY(1)#0 -> main.c:M:ANY(1)#3
 EDGES|CONTAINS out#0 -> out#7
 EDGES|EVAL_TYPE <includes>:<global>#1 -> T:ANY
 EDGES|EVAL_TYPE <includes>:<global>#2 -> T:ANY
+EDGES|EVAL_TYPE <operator>.assignment#1 -> T:ANY
+EDGES|EVAL_TYPE <operator>.assignment#2 -> T:ANY
+EDGES|EVAL_TYPE <operator>.assignment#3 -> T:ANY
+EDGES|EVAL_TYPE <operator>.assignment#4 -> T:ANY
+EDGES|EVAL_TYPE <operator>.assignment#5 -> T:ANY
+EDGES|EVAL_TYPE <operator>.assignment#6 -> T:ANY
+EDGES|EVAL_TYPE <operator>.notEquals#1 -> T:ANY
+EDGES|EVAL_TYPE <operator>.notEquals#2 -> T:ANY
+EDGES|EVAL_TYPE <operator>.notEquals#3 -> T:ANY
+EDGES|EVAL_TYPE <operator>.notEquals#4 -> T:ANY
+EDGES|EVAL_TYPE <operator>.notEquals#5 -> T:ANY
+EDGES|EVAL_TYPE <operator>.notEquals#6 -> T:ANY
 EDGES|EVAL_TYPE <operator>.postDecrement#1 -> T:ANY
 EDGES|EVAL_TYPE <operator>.postDecrement#2 -> T:ANY
 EDGES|EVAL_TYPE <operator>.postDecrement#3 -> T:ANY
@@ -152,9 +283,11 @@
 EDGES|EVAL_TYPE helper#2 -> T:int
 EDGES|EVAL_TYPE helper#3 -> T:ANY
 EDGES|EVAL_TYPE helper#4 -> T:int
-EDGES|EVAL_TYPE main.c:<global>#17 -> T:ANY
-EDGES|EVAL_TYPE main.c:<global>#18 -> T:test
-EDGES|EVAL_TYPE main.c:<global>#19 -> T:ANY
+EDGES|EVAL_TYPE main.c:<global>#36 -> T:ANY
+EDGES|EVAL_TYPE main.c:<global>#37 -> T:test
+EDGES|EVAL_TYPE main.c:<global>#38 -> T:ANY
+EDGES|EVAL_TYPE main.c:F:char*(0)#1 -> T:ANY
+EDGES|EVAL_TYPE main.c:F:char*(0)#2 -> T:char*
 EDGES|EVAL_TYPE main.c:M:ANY(1)#1 -> T:ANY
 EDGES|EVAL_TYPE main.c:M:ANY(1)#2 -> T:ANY
 EDGES|EVAL_TYPE main.c:M:ANY(1)#3 -> T:ANY
@@ -168,18 +301,44 @@
 EDGES|EVAL_TYPE out#7 -> T:ANY
 EDGES|EVAL_TYPE out#8 -> T:int
 EDGES|EVAL_TYPE test#1 -> T:int
-EDGES|EVAL_TYPE test#10 -> T:ANY
-EDGES|EVAL_TYPE test#11 -> T:int
+EDGES|EVAL_TYPE test#10 -> T:int
+EDGES|EVAL_TYPE test#11 -> T:char*
 EDGES|EVAL_TYPE test#12 -> T:ANY
-EDGES|EVAL_TYPE test#13 -> T:int
-EDGES|EVAL_TYPE test#14 -> T:int
+EDGES|EVAL_TYPE test#13 -> T:char*
+EDGES|EVAL_TYPE test#14 -> T:ANY
 EDGES|EVAL_TYPE test#15 -> T:int
+EDGES|EVAL_TYPE test#16 -> T:ANY
+EDGES|EVAL_TYPE test#17 -> T:int
+EDGES|EVAL_TYPE test#18 -> T:int
+EDGES|EVAL_TYPE test#19 -> T:int
 EDGES|EVAL_TYPE test#2 -> T:int
+EDGES|EVAL_TYPE test#20 -> T:int
+EDGES|EVAL_TYPE test#21 -> T:int
+EDGES|EVAL_TYPE test#22 -> T:ANY
+EDGES|EVAL_TYPE test#23 -> T:int
+EDGES|EVAL_TYPE test#24 -> T:void
+EDGES|EVAL_TYPE test#25 -> T:ANY
+EDGES|EVAL_TYPE test#26 -> T:int
+EDGES|EVAL_TYPE test#27 -> T:int
+EDGES|EVAL_TYPE test#29 -> T:ANY
 EDGES|EVAL_TYPE test#3 -> T:void
+EDGES|EVAL_TYPE test#30 -> T:int
+EDGES|EVAL_TYPE test#31 -> T:ANY
+EDGES|EVAL_TYPE test#32 -> T:int
+EDGES|EVAL_TYPE test#33 -> T:int
+EDGES|EVAL_TYPE test#34 -> T:int
 EDGES|EVAL_TYPE test#4 -> T:ANY
-EDGES|EVAL_TYPE test#6 -> T:ANY
-EDGES|EVAL_TYPE test#7 -> T:ANY
+EDGES|EVAL_TYPE test#6 -> T:int
+EDGES|EVAL_TYPE test#7 -> T:void
 EDGES|EVAL_TYPE test#8 -> T:int
+EDGES|EVAL_TYPE test#9 -> T:int
+EDGES|FOR_BODY test#5 -> test#24
+EDGES|FOR_INIT test#5 -> test#7
+EDGES|FOR_UPDATE test#5 -> test#22
+EDGES|PARAMETER_LINK <operator>.assignment#1 -> <operator>.assignment#3
+EDGES|PARAMETER_LINK <operator>.assignment#4 -> <operator>.assignment#6
+EDGES|PARAMETER_LINK <operator>.notEquals#1 -> <operator>.notEquals#3
+EDGES|PARAMETER_LINK <operator>.notEquals#4 -> <operator>.notEquals#6
 EDGES|PARAMETER_LINK <operator>.postDecrement#1 -> <operator>.postDecrement#3
 EDGES|PARAMETER_LINK helper#1 -> helper#2
 EDGES|PARAMETER_LINK main.c:M:ANY(1)#1 -> main.c:M:ANY(1)#2
@@ -195,10 +354,15 @@
 EDGES|REF T:int -> D:int
 EDGES|REF T:test -> D:test
 EDGES|REF T:void -> D:void
-EDGES|REF main.c:<global>#18 -> main.c:<global>#1
-EDGES|REF test#14 -> test#1
-EDGES|REF test#6 -> test#4
-EDGES|REF test#8 -> test#1
+EDGES|REF main.c:<global>#37 -> main.c:<global>#1
+EDGES|REF test#10 -> test#1
+EDGES|REF test#18 -> test#1
+EDGES|REF test#20 -> test#6
+EDGES|REF test#23 -> test#1
+EDGES|REF test#26 -> test#1
+EDGES|REF test#27 -> test#6
+EDGES|REF test#33 -> test#1
+EDGES|REF test#8 -> test#6
 EDGES|SOURCE_FILE <includes>:<global>#0 -> F:<includes>
 EDGES|SOURCE_FILE D:ANY -> F:<includes>
 EDGES|SOURCE_FILE D:char* -> F:<includes>
@@ -212,21 +376,42 @@
 EDGES|SOURCE_FILE helper#0 -> F:main.c
 EDGES|SOURCE_FILE main.c:<global>#0 -> F:main.c
 EDGES|SOURCE_FILE main.c:<global>#1 -> F:main.c
+EDGES|SOURCE_FILE main.c:F:char*(0)#0 -> F:main.c
 EDGES|SOURCE_FILE main.c:M:ANY(1)#0 -> F:main.c
 EDGES|SOURCE_FILE out#0 -> F:main.c
-FLOWS|REACHING_DEF[<RET>] test#9 -> test#15
-FLOWS|REACHING_DEF[M(x)] test#10 -> test#15
-FLOWS|REACHING_DEF[M(x)] test#10 -> test#9
+FLOWS|REACHING_DEF[0] test#21 -> test#19
+FLOWS|REACHING_DEF[0] test#21 -> test#20
+FLOWS|REACHING_DEF[<RET>] test#28 -> test#34
+FLOWS|REACHING_DEF[F] test#11 -> test#10
+FLOWS|REACHING_DEF[F] test#11 -> test#14
+FLOWS|REACHING_DEF[F] test#11 -> test#34
+FLOWS|REACHING_DEF[F] test#11 -> test#9
+FLOWS|REACHING_DEF[M(x)] test#14 -> test#10
+FLOWS|REACHING_DEF[M(x)] test#14 -> test#11
+FLOWS|REACHING_DEF[M(x)] test#14 -> test#9
+FLOWS|REACHING_DEF[M(x)] test#29 -> test#28
+FLOWS|REACHING_DEF[M(x)] test#29 -> test#34
+FLOWS|REACHING_DEF[] <operator>.assignment#0 -> <operator>.assignment#1
+FLOWS|REACHING_DEF[] <operator>.assignment#0 -> <operator>.assignment#4
+FLOWS|REACHING_DEF[] <operator>.notEquals#0 -> <operator>.notEquals#1
+FLOWS|REACHING_DEF[] <operator>.notEquals#0 -> <operator>.notEquals#4
 FLOWS|REACHING_DEF[] <operator>.postDecrement#0 -> <operator>.postDecrement#1
 FLOWS|REACHING_DEF[] helper#0 -> helper#1
 FLOWS|REACHING_DEF[] helper#1 -> helper#2
 FLOWS|REACHING_DEF[] helper#1 -> helper#4
-FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#18
+FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#37
 FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#1
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#10
 FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#11
-FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#14
-FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#6
-FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#8
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#13
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#15
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#18
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#20
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#21
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#23
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#27
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#30
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#33
 FLOWS|REACHING_DEF[] main.c:M:ANY(1)#0 -> main.c:M:ANY(1)#1
 FLOWS|REACHING_DEF[] out#0 -> out#1
 FLOWS|REACHING_DEF[] out#0 -> out#3
@@ -237,20 +422,54 @@
 FLOWS|REACHING_DEF[] out#3 -> out#8
 FLOWS|REACHING_DEF[] out#5 -> out#6
 FLOWS|REACHING_DEF[] out#5 -> out#8
-FLOWS|REACHING_DEF[helper(x)] test#13 -> test#15
+FLOWS|REACHING_DEF[helper(x)] test#17 -> test#34
+FLOWS|REACHING_DEF[helper(x)] test#32 -> test#34
+FLOWS|REACHING_DEF[out(x,F,M(x))] test#9 -> test#34
+FLOWS|REACHING_DEF[out(x,F,M(x))] test#9 -> test#7
+FLOWS|REACHING_DEF[out(x,F,M(x))] test#9 -> test#8
+FLOWS|REACHING_DEF[p1] <operator>.assignment#1 -> <operator>.assignment#3
+FLOWS|REACHING_DEF[p1] <operator>.assignment#1 -> <operator>.assignment#5
+FLOWS|REACHING_DEF[p1] <operator>.notEquals#1 -> <operator>.notEquals#3
+FLOWS|REACHING_DEF[p1] <operator>.notEquals#1 -> <operator>.notEquals#5
 FLOWS|REACHING_DEF[p1] <operator>.postDecrement#1 -> <operator>.postDecrement#3
 FLOWS|REACHING_DEF[p1] <operator>.postDecrement#1 -> <operator>.postDecrement#4
 FLOWS|REACHING_DEF[p1] main.c:M:ANY(1)#1 -> main.c:M:ANY(1)#2
 FLOWS|REACHING_DEF[p1] main.c:M:ANY(1)#1 -> main.c:M:ANY(1)#4
-FLOWS|REACHING_DEF[x--] test#7 -> test#15
+FLOWS|REACHING_DEF[p2] <operator>.assignment#4 -> <operator>.assignment#5
+FLOWS|REACHING_DEF[p2] <operator>.assignment#4 -> <operator>.assignment#6
+FLOWS|REACHING_DEF[p2] <operator>.notEquals#4 -> <operator>.notEquals#5
+FLOWS|REACHING_DEF[p2] <operator>.notEquals#4 -> <operator>.notEquals#6
+FLOWS|REACHING_DEF[x--] test#22 -> test#34
+FLOWS|REACHING_DEF[x=y] test#25 -> test#34
+FLOWS|REACHING_DEF[x] test#1 -> test#15
 FLOWS|REACHING_DEF[x] test#1 -> test#2
-FLOWS|REACHING_DEF[x] test#1 -> test#8
-FLOWS|REACHING_DEF[x] test#11 -> test#10
-FLOWS|REACHING_DEF[x] test#11 -> test#14
-FLOWS|REACHING_DEF[x] test#11 -> test#15
-FLOWS|REACHING_DEF[x] test#11 -> test#2
-FLOWS|REACHING_DEF[x] test#14 -> test#13
-FLOWS|REACHING_DEF[x] test#14 -> test#15
-FLOWS|REACHING_DEF[x] test#14 -> test#2
-FLOWS|REACHING_DEF[x] test#8 -> test#11
-FLOWS|REACHING_DEF[x] test#8 -> test#7
+FLOWS|REACHING_DEF[x] test#10 -> test#11
+FLOWS|REACHING_DEF[x] test#10 -> test#14
+FLOWS|REACHING_DEF[x] test#10 -> test#30
+FLOWS|REACHING_DEF[x] test#10 -> test#9
+FLOWS|REACHING_DEF[x] test#15 -> test#10
+FLOWS|REACHING_DEF[x] test#15 -> test#14
+FLOWS|REACHING_DEF[x] test#15 -> test#18
+FLOWS|REACHING_DEF[x] test#18 -> test#10
+FLOWS|REACHING_DEF[x] test#18 -> test#17
+FLOWS|REACHING_DEF[x] test#23 -> test#22
+FLOWS|REACHING_DEF[x] test#23 -> test#30
+FLOWS|REACHING_DEF[x] test#26 -> test#23
+FLOWS|REACHING_DEF[x] test#26 -> test#25
+FLOWS|REACHING_DEF[x] test#30 -> test#2
+FLOWS|REACHING_DEF[x] test#30 -> test#29
+FLOWS|REACHING_DEF[x] test#30 -> test#33
+FLOWS|REACHING_DEF[x] test#30 -> test#34
+FLOWS|REACHING_DEF[x] test#33 -> test#2
+FLOWS|REACHING_DEF[x] test#33 -> test#32
+FLOWS|REACHING_DEF[x] test#33 -> test#34
+FLOWS|REACHING_DEF[y != 0] test#19 -> test#34
+FLOWS|REACHING_DEF[y=out(x,F,M(x))] test#7 -> test#34
+FLOWS|REACHING_DEF[y] test#20 -> test#19
+FLOWS|REACHING_DEF[y] test#20 -> test#27
+FLOWS|REACHING_DEF[y] test#20 -> test#34
+FLOWS|REACHING_DEF[y] test#27 -> test#20
+FLOWS|REACHING_DEF[y] test#27 -> test#25
+FLOWS|REACHING_DEF[y] test#27 -> test#26
+FLOWS|REACHING_DEF[y] test#8 -> test#20
+FLOWS|REACHING_DEF[y] test#8 -> test#7
