--- joern
+++ before
@@ -16,8 +16,9 @@
       LOCAL NAME=int CODE=int TYPE_FULL_NAME=int ORDER=0
       RETURN CODE=return S(value); ORDER=1
         CALL NAME=S CODE=S(value) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:S:ANY(1) SIGNATURE=ANY(1) ORDER=1 DISPATCH_TYPE=INLINED
-          BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
-            CALL NAME=<operator>.sizeOf CODE=sizeof (int []) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.sizeOf ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
+          IDENTIFIER NAME=value CODE=value TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+          BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
+            CALL NAME=<operator>.sizeOf CODE=sizeof(int[value + 1]) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.sizeOf ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
               IDENTIFIER NAME=int CODE=int TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
     METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
   BLOCK TYPE_FULL_NAME=ANY ORDER=1
@@ -37,8 +38,9 @@
     LOCAL NAME=int CODE=int TYPE_FULL_NAME=int ORDER=0
     RETURN CODE=return S(value); ORDER=1
       CALL NAME=S CODE=S(value) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:S:ANY(1) SIGNATURE=ANY(1) ORDER=1 DISPATCH_TYPE=INLINED
-        BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
-          CALL NAME=<operator>.sizeOf CODE=sizeof (int []) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.sizeOf ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
+        IDENTIFIER NAME=value CODE=value TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+        BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
+          CALL NAME=<operator>.sizeOf CODE=sizeof(int[value + 1]) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.sizeOf ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
             IDENTIFIER NAME=int CODE=int TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
 
@@ -60,20 +62,22 @@
 NODES|TYPE NAME=read FULL_NAME=read TYPE_DECL_FULL_NAME=read
 NODES|TYPE NAME=void FULL_NAME=void TYPE_DECL_FULL_NAME=void
 EDGES|ARGUMENT read#5 -> read#6
-EDGES|ARGUMENT read#8 -> read#9
+EDGES|ARGUMENT read#6 -> read#7
+EDGES|ARGUMENT read#9 -> read#10
 EDGES|CALL read#6 -> main.c:S:ANY(1)#0
-EDGES|CALL read#8 -> <operator>.sizeOf#0
+EDGES|CALL read#9 -> <operator>.sizeOf#0
 EDGES|CFG <includes>:<global>#0 -> <includes>:<global>#2
 EDGES|CFG <operator>.sizeOf#0 -> <operator>.sizeOf#4
-EDGES|CFG main.c:<global>#0 -> main.c:<global>#13
-EDGES|CFG main.c:<global>#1 -> read#6
-EDGES|CFG main.c:<global>#13 -> main.c:<global>#14
+EDGES|CFG main.c:<global>#0 -> main.c:<global>#14
+EDGES|CFG main.c:<global>#1 -> read#7
+EDGES|CFG main.c:<global>#14 -> main.c:<global>#15
 EDGES|CFG main.c:S:ANY(1)#0 -> main.c:S:ANY(1)#4
-EDGES|CFG read#5 -> read#10
+EDGES|CFG read#10 -> read#9
+EDGES|CFG read#5 -> read#11
+EDGES|CFG read#6 -> read#10
 EDGES|CFG read#6 -> read#5
-EDGES|CFG read#6 -> read#9
-EDGES|CFG read#8 -> read#5
-EDGES|CFG read#9 -> read#8
+EDGES|CFG read#7 -> read#6
+EDGES|CFG read#9 -> read#5
 EDGES|CONTAINS <includes>:<global>#0 -> <includes>:<global>#1
 EDGES|CONTAINS <operator>.sizeOf#0 -> <operator>.sizeOf#2
 EDGES|CONTAINS D:main.c:<global> -> D:read
@@ -85,8 +89,9 @@
 EDGES|CONTAINS F:<includes> -> D:void
 EDGES|CONTAINS F:main.c -> D:main.c:<global>
 EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#1
-EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#12
 EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#13
+EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#14
+EDGES|CONTAINS main.c:<global>#1 -> read#10
 EDGES|CONTAINS main.c:<global>#1 -> read#3
 EDGES|CONTAINS main.c:<global>#1 -> read#5
 EDGES|CONTAINS main.c:<global>#1 -> read#6
@@ -100,22 +105,23 @@
 EDGES|EVAL_TYPE <operator>.sizeOf#2 -> T:ANY
 EDGES|EVAL_TYPE <operator>.sizeOf#3 -> T:ANY
 EDGES|EVAL_TYPE <operator>.sizeOf#4 -> T:ANY
-EDGES|EVAL_TYPE main.c:<global>#12 -> T:ANY
-EDGES|EVAL_TYPE main.c:<global>#13 -> T:read
-EDGES|EVAL_TYPE main.c:<global>#14 -> T:ANY
+EDGES|EVAL_TYPE main.c:<global>#13 -> T:ANY
+EDGES|EVAL_TYPE main.c:<global>#14 -> T:read
+EDGES|EVAL_TYPE main.c:<global>#15 -> T:ANY
 EDGES|EVAL_TYPE main.c:S:ANY(1)#1 -> T:ANY
 EDGES|EVAL_TYPE main.c:S:ANY(1)#2 -> T:ANY
 EDGES|EVAL_TYPE main.c:S:ANY(1)#3 -> T:ANY
 EDGES|EVAL_TYPE main.c:S:ANY(1)#4 -> T:ANY
 EDGES|EVAL_TYPE read#1 -> T:int
 EDGES|EVAL_TYPE read#10 -> T:int
+EDGES|EVAL_TYPE read#11 -> T:int
 EDGES|EVAL_TYPE read#2 -> T:int
 EDGES|EVAL_TYPE read#3 -> T:void
 EDGES|EVAL_TYPE read#4 -> T:int
 EDGES|EVAL_TYPE read#6 -> T:ANY
-EDGES|EVAL_TYPE read#7 -> T:ANY
+EDGES|EVAL_TYPE read#7 -> T:int
 EDGES|EVAL_TYPE read#8 -> T:ANY
-EDGES|EVAL_TYPE read#9 -> T:int
+EDGES|EVAL_TYPE read#9 -> T:ANY
 EDGES|PARAMETER_LINK <operator>.sizeOf#1 -> <operator>.sizeOf#3
 EDGES|PARAMETER_LINK main.c:S:ANY(1)#1 -> main.c:S:ANY(1)#2
 EDGES|PARAMETER_LINK read#1 -> read#2
@@ -126,8 +132,8 @@
 EDGES|REF T:int -> D:int
 EDGES|REF T:read -> D:read
 EDGES|REF T:void -> D:void
-EDGES|REF main.c:<global>#13 -> main.c:<global>#1
-EDGES|REF read#9 -> read#4
+EDGES|REF main.c:<global>#14 -> main.c:<global>#1
+EDGES|REF read#10 -> read#4
 EDGES|SOURCE_FILE <includes>:<global>#0 -> F:<includes>
 EDGES|SOURCE_FILE D:ANY -> F:<includes>
 EDGES|SOURCE_FILE D:int -> F:<includes>
@@ -140,17 +146,20 @@
 EDGES|SOURCE_FILE main.c:<global>#0 -> F:main.c
 EDGES|SOURCE_FILE main.c:<global>#1 -> F:main.c
 EDGES|SOURCE_FILE main.c:S:ANY(1)#0 -> F:main.c
-FLOWS|REACHING_DEF[<RET>] read#5 -> read#10
-FLOWS|REACHING_DEF[S(value)] read#6 -> read#10
+FLOWS|REACHING_DEF[<RET>] read#5 -> read#11
+FLOWS|REACHING_DEF[S(value)] read#6 -> read#11
 FLOWS|REACHING_DEF[S(value)] read#6 -> read#5
 FLOWS|REACHING_DEF[] <operator>.sizeOf#0 -> <operator>.sizeOf#1
-FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#13
+FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#14
 FLOWS|REACHING_DEF[] main.c:<global>#1 -> read#1
-FLOWS|REACHING_DEF[] main.c:<global>#1 -> read#6
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> read#7
 FLOWS|REACHING_DEF[] main.c:S:ANY(1)#0 -> main.c:S:ANY(1)#1
 FLOWS|REACHING_DEF[p1] <operator>.sizeOf#1 -> <operator>.sizeOf#3
 FLOWS|REACHING_DEF[p1] <operator>.sizeOf#1 -> <operator>.sizeOf#4
 FLOWS|REACHING_DEF[p1] main.c:S:ANY(1)#1 -> main.c:S:ANY(1)#2
 FLOWS|REACHING_DEF[p1] main.c:S:ANY(1)#1 -> main.c:S:ANY(1)#4
-FLOWS|REACHING_DEF[value] read#1 -> read#10
 FLOWS|REACHING_DEF[value] read#1 -> read#2
+FLOWS|REACHING_DEF[value] read#1 -> read#7
+FLOWS|REACHING_DEF[value] read#7 -> read#11
+FLOWS|REACHING_DEF[value] read#7 -> read#2
+FLOWS|REACHING_DEF[value] read#7 -> read#6
