--- joern
+++ before
@@ -33,7 +33,6 @@
       CALL NAME=<operator>.assignment CODE=first = sizeof(/*inside*/ (( value ))) TYPE_FULL_NAME=void METHOD_FULL_NAME=<operator>.assignment ORDER=2 DISPATCH_TYPE=STATIC_DISPATCH
         IDENTIFIER NAME=first CODE=first TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
         CALL NAME=<operator>.sizeOf CODE=sizeof(/*inside*/ (( value ))) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.sizeOf ORDER=2 ARGUMENT_INDEX=2 DISPATCH_TYPE=STATIC_DISPATCH
-          IDENTIFIER NAME=value CODE=value TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
       LOCAL NAME=second CODE=int second TYPE_FULL_NAME=int ORDER=3
       CALL NAME=<operator>.assignment CODE=second = sizeof sizeof value TYPE_FULL_NAME=void METHOD_FULL_NAME=<operator>.assignment ORDER=4 DISPATCH_TYPE=STATIC_DISPATCH
         IDENTIFIER NAME=second CODE=second TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
@@ -57,7 +56,6 @@
     CALL NAME=<operator>.assignment CODE=first = sizeof(/*inside*/ (( value ))) TYPE_FULL_NAME=void METHOD_FULL_NAME=<operator>.assignment ORDER=2 DISPATCH_TYPE=STATIC_DISPATCH
       IDENTIFIER NAME=first CODE=first TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
       CALL NAME=<operator>.sizeOf CODE=sizeof(/*inside*/ (( value ))) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.sizeOf ORDER=2 ARGUMENT_INDEX=2 DISPATCH_TYPE=STATIC_DISPATCH
-        IDENTIFIER NAME=value CODE=value TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
     LOCAL NAME=second CODE=int second TYPE_FULL_NAME=int ORDER=3
     CALL NAME=<operator>.assignment CODE=second = sizeof sizeof value TYPE_FULL_NAME=void METHOD_FULL_NAME=<operator>.assignment ORDER=4 DISPATCH_TYPE=STATIC_DISPATCH
       IDENTIFIER NAME=second CODE=second TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
@@ -87,42 +85,40 @@
 NODES|TYPE NAME=int FULL_NAME=int TYPE_DECL_FULL_NAME=int
 NODES|TYPE NAME=read FULL_NAME=read TYPE_DECL_FULL_NAME=read
 NODES|TYPE NAME=void FULL_NAME=void TYPE_DECL_FULL_NAME=void
-EDGES|ARGUMENT read#10 -> read#11
-EDGES|ARGUMENT read#10 -> read#12
+EDGES|ARGUMENT read#11 -> read#12
 EDGES|ARGUMENT read#12 -> read#13
-EDGES|ARGUMENT read#13 -> read#14
+EDGES|ARGUMENT read#14 -> read#15
 EDGES|ARGUMENT read#15 -> read#16
-EDGES|ARGUMENT read#16 -> read#17
-EDGES|ARGUMENT read#16 -> read#18
+EDGES|ARGUMENT read#15 -> read#17
 EDGES|ARGUMENT read#5 -> read#6
 EDGES|ARGUMENT read#5 -> read#7
-EDGES|ARGUMENT read#7 -> read#8
-EDGES|CALL read#10 -> <operator>.assignment#0
+EDGES|ARGUMENT read#9 -> read#10
+EDGES|ARGUMENT read#9 -> read#11
+EDGES|CALL read#11 -> <operator>.sizeOf#0
 EDGES|CALL read#12 -> <operator>.sizeOf#0
-EDGES|CALL read#13 -> <operator>.sizeOf#0
-EDGES|CALL read#16 -> <operator>.addition#0
+EDGES|CALL read#15 -> <operator>.addition#0
 EDGES|CALL read#5 -> <operator>.assignment#0
 EDGES|CALL read#7 -> <operator>.sizeOf#0
+EDGES|CALL read#9 -> <operator>.assignment#0
 EDGES|CFG <includes>:<global>#0 -> <includes>:<global>#2
 EDGES|CFG <operator>.addition#0 -> <operator>.addition#5
 EDGES|CFG <operator>.assignment#0 -> <operator>.assignment#5
 EDGES|CFG <operator>.sizeOf#0 -> <operator>.sizeOf#4
-EDGES|CFG main.c:<global>#0 -> main.c:<global>#22
+EDGES|CFG main.c:<global>#0 -> main.c:<global>#21
 EDGES|CFG main.c:<global>#1 -> read#6
-EDGES|CFG main.c:<global>#22 -> main.c:<global>#23
-EDGES|CFG read#10 -> read#17
-EDGES|CFG read#11 -> read#14
-EDGES|CFG read#12 -> read#10
+EDGES|CFG main.c:<global>#21 -> main.c:<global>#22
+EDGES|CFG read#10 -> read#13
+EDGES|CFG read#11 -> read#9
+EDGES|CFG read#12 -> read#11
 EDGES|CFG read#13 -> read#12
-EDGES|CFG read#14 -> read#13
-EDGES|CFG read#15 -> read#19
-EDGES|CFG read#16 -> read#15
-EDGES|CFG read#17 -> read#18
-EDGES|CFG read#18 -> read#16
-EDGES|CFG read#5 -> read#11
-EDGES|CFG read#6 -> read#8
+EDGES|CFG read#14 -> read#18
+EDGES|CFG read#15 -> read#14
+EDGES|CFG read#16 -> read#17
+EDGES|CFG read#17 -> read#15
+EDGES|CFG read#5 -> read#10
+EDGES|CFG read#6 -> read#7
 EDGES|CFG read#7 -> read#5
-EDGES|CFG read#8 -> read#7
+EDGES|CFG read#9 -> read#16
 EDGES|CONTAINS <includes>:<global>#0 -> <includes>:<global>#1
 EDGES|CONTAINS <operator>.addition#0 -> <operator>.addition#2
 EDGES|CONTAINS <operator>.assignment#0 -> <operator>.assignment#2
@@ -135,8 +131,8 @@
 EDGES|CONTAINS F:<includes> -> D:void
 EDGES|CONTAINS F:main.c -> D:main.c:<global>
 EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#1
+EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#20
 EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#21
-EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#22
 EDGES|CONTAINS main.c:<global>#1 -> read#10
 EDGES|CONTAINS main.c:<global>#1 -> read#11
 EDGES|CONTAINS main.c:<global>#1 -> read#12
@@ -145,12 +141,11 @@
 EDGES|CONTAINS main.c:<global>#1 -> read#15
 EDGES|CONTAINS main.c:<global>#1 -> read#16
 EDGES|CONTAINS main.c:<global>#1 -> read#17
-EDGES|CONTAINS main.c:<global>#1 -> read#18
 EDGES|CONTAINS main.c:<global>#1 -> read#3
 EDGES|CONTAINS main.c:<global>#1 -> read#5
 EDGES|CONTAINS main.c:<global>#1 -> read#6
 EDGES|CONTAINS main.c:<global>#1 -> read#7
-EDGES|CONTAINS main.c:<global>#1 -> read#8
+EDGES|CONTAINS main.c:<global>#1 -> read#9
 EDGES|EVAL_TYPE <includes>:<global>#1 -> T:ANY
 EDGES|EVAL_TYPE <includes>:<global>#2 -> T:ANY
 EDGES|EVAL_TYPE <operator>.addition#1 -> T:ANY
@@ -169,19 +164,18 @@
 EDGES|EVAL_TYPE <operator>.sizeOf#2 -> T:ANY
 EDGES|EVAL_TYPE <operator>.sizeOf#3 -> T:ANY
 EDGES|EVAL_TYPE <operator>.sizeOf#4 -> T:ANY
-EDGES|EVAL_TYPE main.c:<global>#21 -> T:ANY
-EDGES|EVAL_TYPE main.c:<global>#22 -> T:read
-EDGES|EVAL_TYPE main.c:<global>#23 -> T:ANY
+EDGES|EVAL_TYPE main.c:<global>#20 -> T:ANY
+EDGES|EVAL_TYPE main.c:<global>#21 -> T:read
+EDGES|EVAL_TYPE main.c:<global>#22 -> T:ANY
 EDGES|EVAL_TYPE read#1 -> T:int
-EDGES|EVAL_TYPE read#10 -> T:void
-EDGES|EVAL_TYPE read#11 -> T:int
+EDGES|EVAL_TYPE read#10 -> T:int
+EDGES|EVAL_TYPE read#11 -> T:ANY
 EDGES|EVAL_TYPE read#12 -> T:ANY
-EDGES|EVAL_TYPE read#13 -> T:ANY
-EDGES|EVAL_TYPE read#14 -> T:int
-EDGES|EVAL_TYPE read#16 -> T:ANY
+EDGES|EVAL_TYPE read#13 -> T:int
+EDGES|EVAL_TYPE read#15 -> T:ANY
+EDGES|EVAL_TYPE read#16 -> T:int
 EDGES|EVAL_TYPE read#17 -> T:int
 EDGES|EVAL_TYPE read#18 -> T:int
-EDGES|EVAL_TYPE read#19 -> T:int
 EDGES|EVAL_TYPE read#2 -> T:int
 EDGES|EVAL_TYPE read#3 -> T:void
 EDGES|EVAL_TYPE read#4 -> T:int
@@ -189,7 +183,7 @@
 EDGES|EVAL_TYPE read#6 -> T:int
 EDGES|EVAL_TYPE read#7 -> T:ANY
 EDGES|EVAL_TYPE read#8 -> T:int
-EDGES|EVAL_TYPE read#9 -> T:int
+EDGES|EVAL_TYPE read#9 -> T:void
 EDGES|PARAMETER_LINK <operator>.addition#1 -> <operator>.addition#3
 EDGES|PARAMETER_LINK <operator>.addition#4 -> <operator>.addition#6
 EDGES|PARAMETER_LINK <operator>.assignment#1 -> <operator>.assignment#3
@@ -203,13 +197,12 @@
 EDGES|REF T:int -> D:int
 EDGES|REF T:read -> D:read
 EDGES|REF T:void -> D:void
-EDGES|REF main.c:<global>#22 -> main.c:<global>#1
-EDGES|REF read#11 -> read#9
-EDGES|REF read#14 -> read#1
-EDGES|REF read#17 -> read#4
-EDGES|REF read#18 -> read#9
+EDGES|REF main.c:<global>#21 -> main.c:<global>#1
+EDGES|REF read#10 -> read#8
+EDGES|REF read#13 -> read#1
+EDGES|REF read#16 -> read#4
+EDGES|REF read#17 -> read#8
 EDGES|REF read#6 -> read#4
-EDGES|REF read#8 -> read#1
 EDGES|SOURCE_FILE <includes>:<global>#0 -> F:<includes>
 EDGES|SOURCE_FILE D:ANY -> F:<includes>
 EDGES|SOURCE_FILE D:int -> F:<includes>
@@ -221,22 +214,23 @@
 EDGES|SOURCE_FILE NB:main.c:<global> -> F:main.c
 EDGES|SOURCE_FILE main.c:<global>#0 -> F:main.c
 EDGES|SOURCE_FILE main.c:<global>#1 -> F:main.c
-FLOWS|REACHING_DEF[<RET>] read#15 -> read#19
+FLOWS|REACHING_DEF[<RET>] read#14 -> read#18
 FLOWS|REACHING_DEF[] <operator>.addition#0 -> <operator>.addition#1
 FLOWS|REACHING_DEF[] <operator>.addition#0 -> <operator>.addition#4
 FLOWS|REACHING_DEF[] <operator>.assignment#0 -> <operator>.assignment#1
 FLOWS|REACHING_DEF[] <operator>.assignment#0 -> <operator>.assignment#4
 FLOWS|REACHING_DEF[] <operator>.sizeOf#0 -> <operator>.sizeOf#1
-FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#22
+FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#21
 FLOWS|REACHING_DEF[] main.c:<global>#1 -> read#1
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> read#16
 FLOWS|REACHING_DEF[] main.c:<global>#1 -> read#17
-FLOWS|REACHING_DEF[] main.c:<global>#1 -> read#18
-FLOWS|REACHING_DEF[first + second] read#16 -> read#15
-FLOWS|REACHING_DEF[first + second] read#16 -> read#19
-FLOWS|REACHING_DEF[first = sizeof(/*inside*/ (( value )))] read#5 -> read#19
-FLOWS|REACHING_DEF[first] read#17 -> read#16
-FLOWS|REACHING_DEF[first] read#17 -> read#19
-FLOWS|REACHING_DEF[first] read#6 -> read#17
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> read#7
+FLOWS|REACHING_DEF[first + second] read#15 -> read#14
+FLOWS|REACHING_DEF[first + second] read#15 -> read#18
+FLOWS|REACHING_DEF[first = sizeof(/*inside*/ (( value )))] read#5 -> read#18
+FLOWS|REACHING_DEF[first] read#16 -> read#15
+FLOWS|REACHING_DEF[first] read#16 -> read#18
+FLOWS|REACHING_DEF[first] read#6 -> read#16
 FLOWS|REACHING_DEF[first] read#6 -> read#5
 FLOWS|REACHING_DEF[p1] <operator>.addition#1 -> <operator>.addition#3
 FLOWS|REACHING_DEF[p1] <operator>.addition#1 -> <operator>.addition#5
@@ -248,10 +242,10 @@
 FLOWS|REACHING_DEF[p2] <operator>.addition#4 -> <operator>.addition#6
 FLOWS|REACHING_DEF[p2] <operator>.assignment#4 -> <operator>.assignment#5
 FLOWS|REACHING_DEF[p2] <operator>.assignment#4 -> <operator>.assignment#6
-FLOWS|REACHING_DEF[second = sizeof sizeof value] read#10 -> read#19
-FLOWS|REACHING_DEF[second] read#11 -> read#10
-FLOWS|REACHING_DEF[second] read#11 -> read#18
-FLOWS|REACHING_DEF[second] read#18 -> read#16
-FLOWS|REACHING_DEF[second] read#18 -> read#19
-FLOWS|REACHING_DEF[value] read#1 -> read#19
+FLOWS|REACHING_DEF[second = sizeof sizeof value] read#9 -> read#18
+FLOWS|REACHING_DEF[second] read#10 -> read#17
+FLOWS|REACHING_DEF[second] read#10 -> read#9
+FLOWS|REACHING_DEF[second] read#17 -> read#15
+FLOWS|REACHING_DEF[second] read#17 -> read#18
+FLOWS|REACHING_DEF[value] read#1 -> read#18
 FLOWS|REACHING_DEF[value] read#1 -> read#2
