--- joern
+++ before
@@ -3,14 +3,6 @@
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
 
 METHOD NAME=<operator>.assignment FULL_NAME=<operator>.assignment ORDER=0
-  METHOD_PARAMETER_IN NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
-  BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
-  METHOD_PARAMETER_OUT NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
-  METHOD_PARAMETER_IN NAME=p2 CODE=p2 TYPE_FULL_NAME=ANY ORDER=2
-  METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
-  METHOD_PARAMETER_OUT NAME=p2 CODE=p2 TYPE_FULL_NAME=ANY ORDER=2
-
-METHOD NAME=<operator>.cast FULL_NAME=<operator>.cast ORDER=0
   METHOD_PARAMETER_IN NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
   BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
   METHOD_PARAMETER_OUT NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
@@ -48,9 +40,9 @@
         CALL NAME=CAST CODE=CAST(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:CAST:ANY(1) SIGNATURE=ANY(1) ORDER=1 DISPATCH_TYPE=INLINED
           IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
           BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
-            CALL NAME=<operator>.cast CODE=(T)(x) TYPE_FULL_NAME=T METHOD_FULL_NAME=<operator>.cast ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
-              TYPE_REF CODE=T TYPE_FULL_NAME=T ORDER=1 ARGUMENT_INDEX=1
-              IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
+            CALL NAME=<operator>.pointerCall CODE=(T)(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.pointerCall ORDER=1 DISPATCH_TYPE=DYNAMIC_DISPATCH
+              IDENTIFIER NAME=T CODE=<unknown> T TYPE_FULL_NAME=ANY ORDER=1
+              IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=1
     METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
   BLOCK TYPE_FULL_NAME=ANY ORDER=1
     TYPE_DECL NAME=T CODE=typedef int T; FULL_NAME=T ORDER=1
@@ -80,9 +72,9 @@
       CALL NAME=CAST CODE=CAST(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:CAST:ANY(1) SIGNATURE=ANY(1) ORDER=1 DISPATCH_TYPE=INLINED
         IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
         BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
-          CALL NAME=<operator>.cast CODE=(T)(x) TYPE_FULL_NAME=T METHOD_FULL_NAME=<operator>.cast ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
-            TYPE_REF CODE=T TYPE_FULL_NAME=T ORDER=1 ARGUMENT_INDEX=1
-            IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
+          CALL NAME=<operator>.pointerCall CODE=(T)(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.pointerCall ORDER=1 DISPATCH_TYPE=DYNAMIC_DISPATCH
+            IDENTIFIER NAME=T CODE=<unknown> T TYPE_FULL_NAME=ANY ORDER=1
+            IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=1
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
 
 NODES|META_DATA LANGUAGE=NEWC
@@ -107,18 +99,15 @@
 EDGES|ARGUMENT value#10 -> value#12
 EDGES|ARGUMENT value#13 -> value#14
 EDGES|ARGUMENT value#14 -> value#15
-EDGES|ARGUMENT value#17 -> value#18
 EDGES|ARGUMENT value#17 -> value#19
 EDGES|ARGUMENT value#5 -> value#6
 EDGES|ARGUMENT value#5 -> value#7
 EDGES|ARGUMENT value#7 -> value#8
 EDGES|CALL value#14 -> main.c:CAST:ANY(1)#0
-EDGES|CALL value#17 -> <operator>.cast#0
 EDGES|CALL value#5 -> <operator>.assignment#0
 EDGES|CALL value#7 -> main.c:CAST:ANY(1)#0
 EDGES|CFG <includes>:<global>#0 -> <includes>:<global>#2
 EDGES|CFG <operator>.assignment#0 -> <operator>.assignment#5
-EDGES|CFG <operator>.cast#0 -> <operator>.cast#5
 EDGES|CFG <operator>.pointerCall#0 -> <operator>.pointerCall#4
 EDGES|CFG T#0 -> T#4
 EDGES|CFG main.c:<global>#0 -> main.c:<global>#24
@@ -143,7 +132,6 @@
 EDGES|CFG value#8 -> value#7
 EDGES|CONTAINS <includes>:<global>#0 -> <includes>:<global>#1
 EDGES|CONTAINS <operator>.assignment#0 -> <operator>.assignment#2
-EDGES|CONTAINS <operator>.cast#0 -> <operator>.cast#2
 EDGES|CONTAINS <operator>.pointerCall#0 -> <operator>.pointerCall#2
 EDGES|CONTAINS D:main.c:<global> -> D:value
 EDGES|CONTAINS D:main.c:<global> -> T#0
@@ -185,12 +173,6 @@
 EDGES|EVAL_TYPE <operator>.assignment#4 -> T:ANY
 EDGES|EVAL_TYPE <operator>.assignment#5 -> T:ANY
 EDGES|EVAL_TYPE <operator>.assignment#6 -> T:ANY
-EDGES|EVAL_TYPE <operator>.cast#1 -> T:ANY
-EDGES|EVAL_TYPE <operator>.cast#2 -> T:ANY
-EDGES|EVAL_TYPE <operator>.cast#3 -> T:ANY
-EDGES|EVAL_TYPE <operator>.cast#4 -> T:ANY
-EDGES|EVAL_TYPE <operator>.cast#5 -> T:ANY
-EDGES|EVAL_TYPE <operator>.cast#6 -> T:ANY
 EDGES|EVAL_TYPE <operator>.pointerCall#1 -> T:ANY
 EDGES|EVAL_TYPE <operator>.pointerCall#2 -> T:ANY
 EDGES|EVAL_TYPE <operator>.pointerCall#3 -> T:ANY
@@ -213,8 +195,8 @@
 EDGES|EVAL_TYPE value#14 -> T:ANY
 EDGES|EVAL_TYPE value#15 -> T:int
 EDGES|EVAL_TYPE value#16 -> T:ANY
-EDGES|EVAL_TYPE value#17 -> T:T
-EDGES|EVAL_TYPE value#18 -> T:T
+EDGES|EVAL_TYPE value#17 -> T:ANY
+EDGES|EVAL_TYPE value#18 -> T:ANY
 EDGES|EVAL_TYPE value#19 -> T:int
 EDGES|EVAL_TYPE value#2 -> T:int
 EDGES|EVAL_TYPE value#20 -> T:int
@@ -227,8 +209,6 @@
 EDGES|EVAL_TYPE value#9 -> T:ANY
 EDGES|PARAMETER_LINK <operator>.assignment#1 -> <operator>.assignment#3
 EDGES|PARAMETER_LINK <operator>.assignment#4 -> <operator>.assignment#6
-EDGES|PARAMETER_LINK <operator>.cast#1 -> <operator>.cast#3
-EDGES|PARAMETER_LINK <operator>.cast#4 -> <operator>.cast#6
 EDGES|PARAMETER_LINK <operator>.pointerCall#1 -> <operator>.pointerCall#3
 EDGES|PARAMETER_LINK T#1 -> T#2
 EDGES|PARAMETER_LINK main.c:CAST:ANY(1)#1 -> main.c:CAST:ANY(1)#2
@@ -267,11 +247,8 @@
 FLOWS|REACHING_DEF[CAST(x)] value#14 -> value#20
 FLOWS|REACHING_DEF[CAST(x)] value#7 -> value#5
 FLOWS|REACHING_DEF[CAST(x)] value#7 -> value#6
-FLOWS|REACHING_DEF[T] value#18 -> value#17
 FLOWS|REACHING_DEF[] <operator>.assignment#0 -> <operator>.assignment#1
 FLOWS|REACHING_DEF[] <operator>.assignment#0 -> <operator>.assignment#4
-FLOWS|REACHING_DEF[] <operator>.cast#0 -> <operator>.cast#1
-FLOWS|REACHING_DEF[] <operator>.cast#0 -> <operator>.cast#4
 FLOWS|REACHING_DEF[] <operator>.pointerCall#0 -> <operator>.pointerCall#1
 FLOWS|REACHING_DEF[] T#0 -> T#1
 FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#24
@@ -286,16 +263,12 @@
 FLOWS|REACHING_DEF[] main.c:CAST:ANY(1)#0 -> main.c:CAST:ANY(1)#1
 FLOWS|REACHING_DEF[p1] <operator>.assignment#1 -> <operator>.assignment#3
 FLOWS|REACHING_DEF[p1] <operator>.assignment#1 -> <operator>.assignment#5
-FLOWS|REACHING_DEF[p1] <operator>.cast#1 -> <operator>.cast#3
-FLOWS|REACHING_DEF[p1] <operator>.cast#1 -> <operator>.cast#5
 FLOWS|REACHING_DEF[p1] <operator>.pointerCall#1 -> <operator>.pointerCall#3
 FLOWS|REACHING_DEF[p1] <operator>.pointerCall#1 -> <operator>.pointerCall#4
 FLOWS|REACHING_DEF[p1] main.c:CAST:ANY(1)#1 -> main.c:CAST:ANY(1)#2
 FLOWS|REACHING_DEF[p1] main.c:CAST:ANY(1)#1 -> main.c:CAST:ANY(1)#4
 FLOWS|REACHING_DEF[p2] <operator>.assignment#4 -> <operator>.assignment#5
 FLOWS|REACHING_DEF[p2] <operator>.assignment#4 -> <operator>.assignment#6
-FLOWS|REACHING_DEF[p2] <operator>.cast#4 -> <operator>.cast#5
-FLOWS|REACHING_DEF[p2] <operator>.cast#4 -> <operator>.cast#6
 FLOWS|REACHING_DEF[x=CAST(x)] value#5 -> value#20
 FLOWS|REACHING_DEF[x] T#1 -> T#2
 FLOWS|REACHING_DEF[x] T#1 -> T#4
