--- joern
+++ fixed
@@ -1,11 +1,5 @@
 METHOD NAME=<global> CODE=<global> FULL_NAME=<includes>:<global> ORDER=1
   BLOCK TYPE_FULL_NAME=ANY ORDER=1
-  METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
-
-METHOD NAME=<operator>.pointerCall FULL_NAME=<operator>.pointerCall ORDER=0
-  METHOD_PARAMETER_IN NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
-  BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
-  METHOD_PARAMETER_OUT NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
 
 METHOD NAME=f CODE=int f(int x) { return OUT(x); } FULL_NAME=f SIGNATURE=int(int) ORDER=1
@@ -13,12 +7,9 @@
   METHOD_PARAMETER_OUT NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
   BLOCK CODE={ return OUT(x); } TYPE_FULL_NAME=void ORDER=2
     RETURN CODE=return OUT(x); ORDER=1
-      CALL NAME=OUT CODE=OUT(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:OUT:ANY(1) SIGNATURE=ANY(1) ORDER=1 DISPATCH_TYPE=INLINED
-        IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
-        BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
-          CALL NAME=<operator>.pointerCall CODE=1M(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.pointerCall ORDER=1 DISPATCH_TYPE=DYNAMIC_DISPATCH
-            LITERAL CODE=1M TYPE_FULL_NAME=int ORDER=1
-            IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=1
+      CALL NAME=OUT CODE=OUT(x) TYPE_FULL_NAME=int METHOD_FULL_NAME=main.c:OUT:int(1) SIGNATURE=int(1) ORDER=1 DISPATCH_TYPE=INLINED
+        BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
+          LITERAL CODE=1 TYPE_FULL_NAME=int ORDER=1
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
 
 METHOD NAME=<global> CODE=<global> FULL_NAME=main.c:<global> ORDER=1
@@ -27,22 +18,19 @@
     METHOD_PARAMETER_OUT NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
     BLOCK CODE={ return OUT(x); } TYPE_FULL_NAME=void ORDER=2
       RETURN CODE=return OUT(x); ORDER=1
-        CALL NAME=OUT CODE=OUT(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:OUT:ANY(1) SIGNATURE=ANY(1) ORDER=1 DISPATCH_TYPE=INLINED
-          IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
-          BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
-            CALL NAME=<operator>.pointerCall CODE=1M(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.pointerCall ORDER=1 DISPATCH_TYPE=DYNAMIC_DISPATCH
-              LITERAL CODE=1M TYPE_FULL_NAME=int ORDER=1
-              IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=1
+        CALL NAME=OUT CODE=OUT(x) TYPE_FULL_NAME=int METHOD_FULL_NAME=main.c:OUT:int(1) SIGNATURE=int(1) ORDER=1 DISPATCH_TYPE=INLINED
+          BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
+            LITERAL CODE=1 TYPE_FULL_NAME=int ORDER=1
     METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
   BLOCK TYPE_FULL_NAME=ANY ORDER=1
     METHOD_REF CODE=f TYPE_FULL_NAME=f METHOD_FULL_NAME=f ORDER=1
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
 
-METHOD NAME=OUT CODE=#define OUT(x) 1M(x) FULL_NAME=main.c:OUT:ANY(1) SIGNATURE=ANY(1) ORDER=1
+METHOD NAME=OUT CODE=#define OUT(x) 1M(x) FULL_NAME=main.c:OUT:int(1) SIGNATURE=int(1) ORDER=1
   METHOD_PARAMETER_IN NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
   METHOD_PARAMETER_OUT NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
   BLOCK TYPE_FULL_NAME=ANY ORDER=2
-  METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=3
+  METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
 
 NODES|META_DATA LANGUAGE=NEWC
 NODES|FILE NAME=<includes> ORDER=1
@@ -62,70 +50,52 @@
 NODES|TYPE NAME=int FULL_NAME=int TYPE_DECL_FULL_NAME=int
 NODES|TYPE NAME=void FULL_NAME=void TYPE_DECL_FULL_NAME=void
 EDGES|ARGUMENT f#4 -> f#5
-EDGES|ARGUMENT f#5 -> f#6
-EDGES|ARGUMENT f#8 -> f#10
-EDGES|CALL f#5 -> main.c:OUT:ANY(1)#0
+EDGES|CALL f#5 -> main.c:OUT:int(1)#0
 EDGES|CFG <includes>:<global>#0 -> <includes>:<global>#2
-EDGES|CFG <operator>.pointerCall#0 -> <operator>.pointerCall#4
-EDGES|CFG f#0 -> f#6
-EDGES|CFG f#10 -> f#8
-EDGES|CFG f#4 -> f#11
+EDGES|CFG f#0 -> f#5
+EDGES|CFG f#4 -> f#8
 EDGES|CFG f#5 -> f#4
-EDGES|CFG f#5 -> f#9
-EDGES|CFG f#6 -> f#5
-EDGES|CFG f#8 -> f#4
-EDGES|CFG f#9 -> f#10
-EDGES|CFG main.c:<global>#0 -> main.c:<global>#14
-EDGES|CFG main.c:<global>#14 -> main.c:<global>#15
-EDGES|CFG main.c:OUT:ANY(1)#0 -> main.c:OUT:ANY(1)#4
+EDGES|CFG f#5 -> f#7
+EDGES|CFG f#7 -> f#4
+EDGES|CFG main.c:<global>#0 -> main.c:<global>#11
+EDGES|CFG main.c:<global>#11 -> main.c:<global>#12
+EDGES|CFG main.c:OUT:int(1)#0 -> main.c:OUT:int(1)#4
 EDGES|CONTAINS <includes>:<global>#0 -> <includes>:<global>#1
-EDGES|CONTAINS <operator>.pointerCall#0 -> <operator>.pointerCall#2
 EDGES|CONTAINS D:main.c:<global> -> D:f
 EDGES|CONTAINS D:main.c:<global> -> main.c:<global>#0
-EDGES|CONTAINS D:main.c:<global> -> main.c:OUT:ANY(1)#0
+EDGES|CONTAINS D:main.c:<global> -> main.c:OUT:int(1)#0
 EDGES|CONTAINS F:<includes> -> <includes>:<global>#0
 EDGES|CONTAINS F:<includes> -> D:ANY
 EDGES|CONTAINS F:<includes> -> D:int
 EDGES|CONTAINS F:<includes> -> D:void
 EDGES|CONTAINS F:main.c -> D:main.c:<global>
-EDGES|CONTAINS f#0 -> f#10
 EDGES|CONTAINS f#0 -> f#3
 EDGES|CONTAINS f#0 -> f#4
 EDGES|CONTAINS f#0 -> f#5
 EDGES|CONTAINS f#0 -> f#6
 EDGES|CONTAINS f#0 -> f#7
-EDGES|CONTAINS f#0 -> f#8
-EDGES|CONTAINS f#0 -> f#9
 EDGES|CONTAINS main.c:<global>#0 -> f#0
-EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#13
-EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#14
-EDGES|CONTAINS main.c:OUT:ANY(1)#0 -> main.c:OUT:ANY(1)#3
+EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#10
+EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#11
+EDGES|CONTAINS main.c:OUT:int(1)#0 -> main.c:OUT:int(1)#3
 EDGES|EVAL_TYPE <includes>:<global>#1 -> T:ANY
 EDGES|EVAL_TYPE <includes>:<global>#2 -> T:ANY
-EDGES|EVAL_TYPE <operator>.pointerCall#1 -> T:ANY
-EDGES|EVAL_TYPE <operator>.pointerCall#2 -> T:ANY
-EDGES|EVAL_TYPE <operator>.pointerCall#3 -> T:ANY
-EDGES|EVAL_TYPE <operator>.pointerCall#4 -> T:ANY
 EDGES|EVAL_TYPE f#1 -> T:int
-EDGES|EVAL_TYPE f#10 -> T:int
-EDGES|EVAL_TYPE f#11 -> T:int
 EDGES|EVAL_TYPE f#2 -> T:int
 EDGES|EVAL_TYPE f#3 -> T:void
-EDGES|EVAL_TYPE f#5 -> T:ANY
-EDGES|EVAL_TYPE f#6 -> T:int
-EDGES|EVAL_TYPE f#7 -> T:ANY
-EDGES|EVAL_TYPE f#8 -> T:ANY
-EDGES|EVAL_TYPE f#9 -> T:int
-EDGES|EVAL_TYPE main.c:<global>#13 -> T:ANY
-EDGES|EVAL_TYPE main.c:<global>#14 -> T:f
-EDGES|EVAL_TYPE main.c:<global>#15 -> T:ANY
-EDGES|EVAL_TYPE main.c:OUT:ANY(1)#1 -> T:ANY
-EDGES|EVAL_TYPE main.c:OUT:ANY(1)#2 -> T:ANY
-EDGES|EVAL_TYPE main.c:OUT:ANY(1)#3 -> T:ANY
-EDGES|EVAL_TYPE main.c:OUT:ANY(1)#4 -> T:ANY
-EDGES|PARAMETER_LINK <operator>.pointerCall#1 -> <operator>.pointerCall#3
+EDGES|EVAL_TYPE f#5 -> T:int
+EDGES|EVAL_TYPE f#6 -> T:ANY
+EDGES|EVAL_TYPE f#7 -> T:int
+EDGES|EVAL_TYPE f#8 -> T:int
+EDGES|EVAL_TYPE main.c:<global>#10 -> T:ANY
+EDGES|EVAL_TYPE main.c:<global>#11 -> T:f
+EDGES|EVAL_TYPE main.c:<global>#12 -> T:ANY
+EDGES|EVAL_TYPE main.c:OUT:int(1)#1 -> T:ANY
+EDGES|EVAL_TYPE main.c:OUT:int(1)#2 -> T:ANY
+EDGES|EVAL_TYPE main.c:OUT:int(1)#3 -> T:ANY
+EDGES|EVAL_TYPE main.c:OUT:int(1)#4 -> T:int
 EDGES|PARAMETER_LINK f#1 -> f#2
-EDGES|PARAMETER_LINK main.c:OUT:ANY(1)#1 -> main.c:OUT:ANY(1)#2
+EDGES|PARAMETER_LINK main.c:OUT:int(1)#1 -> main.c:OUT:int(1)#2
 EDGES|REF NB:<global> -> NS:<global>
 EDGES|REF NB:<includes>:<global> -> NS:<global>
 EDGES|REF NB:main.c:<global> -> NS:<global>
@@ -133,8 +103,7 @@
 EDGES|REF T:f -> D:f
 EDGES|REF T:int -> D:int
 EDGES|REF T:void -> D:void
-EDGES|REF f#10 -> f#1
-EDGES|REF main.c:<global>#14 -> f#0
+EDGES|REF main.c:<global>#11 -> f#0
 EDGES|SOURCE_FILE <includes>:<global>#0 -> F:<includes>
 EDGES|SOURCE_FILE D:ANY -> F:<includes>
 EDGES|SOURCE_FILE D:f -> F:main.c
@@ -146,28 +115,16 @@
 EDGES|SOURCE_FILE NB:main.c:<global> -> F:main.c
 EDGES|SOURCE_FILE f#0 -> F:main.c
 EDGES|SOURCE_FILE main.c:<global>#0 -> F:main.c
-EDGES|SOURCE_FILE main.c:OUT:ANY(1)#0 -> F:main.c
-FLOWS|REACHING_DEF[1M(x)] f#8 -> f#11
-FLOWS|REACHING_DEF[<RET>] f#4 -> f#11
-FLOWS|REACHING_DEF[OUT(x)] f#5 -> f#11
+EDGES|SOURCE_FILE main.c:OUT:int(1)#0 -> F:main.c
+FLOWS|REACHING_DEF[<RET>] f#4 -> f#8
 FLOWS|REACHING_DEF[OUT(x)] f#5 -> f#4
-FLOWS|REACHING_DEF[] <operator>.pointerCall#0 -> <operator>.pointerCall#1
+FLOWS|REACHING_DEF[OUT(x)] f#5 -> f#8
 FLOWS|REACHING_DEF[] f#0 -> f#1
-FLOWS|REACHING_DEF[] f#0 -> f#10
-FLOWS|REACHING_DEF[] f#0 -> f#6
-FLOWS|REACHING_DEF[] f#0 -> f#9
-FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#14
-FLOWS|REACHING_DEF[] main.c:OUT:ANY(1)#0 -> main.c:OUT:ANY(1)#1
-FLOWS|REACHING_DEF[p1] <operator>.pointerCall#1 -> <operator>.pointerCall#3
-FLOWS|REACHING_DEF[p1] <operator>.pointerCall#1 -> <operator>.pointerCall#4
-FLOWS|REACHING_DEF[p1] main.c:OUT:ANY(1)#1 -> main.c:OUT:ANY(1)#2
-FLOWS|REACHING_DEF[p1] main.c:OUT:ANY(1)#1 -> main.c:OUT:ANY(1)#4
+FLOWS|REACHING_DEF[] f#0 -> f#5
+FLOWS|REACHING_DEF[] f#0 -> f#7
+FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#11
+FLOWS|REACHING_DEF[] main.c:OUT:int(1)#0 -> main.c:OUT:int(1)#1
+FLOWS|REACHING_DEF[p1] main.c:OUT:int(1)#1 -> main.c:OUT:int(1)#2
+FLOWS|REACHING_DEF[p1] main.c:OUT:int(1)#1 -> main.c:OUT:int(1)#4
 FLOWS|REACHING_DEF[x] f#1 -> f#2
-FLOWS|REACHING_DEF[x] f#1 -> f#6
-FLOWS|REACHING_DEF[x] f#10 -> f#11
-FLOWS|REACHING_DEF[x] f#10 -> f#2
-FLOWS|REACHING_DEF[x] f#10 -> f#8
-FLOWS|REACHING_DEF[x] f#6 -> f#10
-FLOWS|REACHING_DEF[x] f#6 -> f#11
-FLOWS|REACHING_DEF[x] f#6 -> f#2
-FLOWS|REACHING_DEF[x] f#6 -> f#5
+FLOWS|REACHING_DEF[x] f#1 -> f#8
