--- joern
+++ before
@@ -61,7 +61,6 @@
 EDGES|CFG f#7 -> f#4
 EDGES|CFG main.c:<global>#0 -> main.c:<global>#11
 EDGES|CFG main.c:<global>#11 -> main.c:<global>#12
-EDGES|CFG main.c:VALUE:unsigned int(1)#0 -> main.c:VALUE:unsigned int(1)#4
 EDGES|CONTAINS <includes>:<global>#0 -> <includes>:<global>#1
 EDGES|CONTAINS D:main.c:<global> -> D:f
 EDGES|CONTAINS D:main.c:<global> -> main.c:<global>#0
@@ -128,7 +127,6 @@
 FLOWS|REACHING_DEF[] f#0 -> f#5
 FLOWS|REACHING_DEF[] f#0 -> f#7
 FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#11
-FLOWS|REACHING_DEF[] main.c:VALUE:unsigned int(1)#0 -> main.c:VALUE:unsigned int(1)#1
 FLOWS|REACHING_DEF[p1] main.c:VALUE:unsigned int(1)#1 -> main.c:VALUE:unsigned int(1)#2
 FLOWS|REACHING_DEF[p1] main.c:VALUE:unsigned int(1)#1 -> main.c:VALUE:unsigned int(1)#4
 FLOWS|REACHING_DEF[x] f#1 -> f#2
