--- Joern
+++ before
@@ -18,12 +18,6 @@
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
   METHOD_PARAMETER_OUT NAME=p2 CODE=p2 TYPE_FULL_NAME=ANY ORDER=2
 
-METHOD NAME=M FULL_NAME=M ORDER=0
-  METHOD_PARAMETER_IN NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
-  BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
-  METHOD_PARAMETER_OUT NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
-  METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
-
 METHOD NAME=helper CODE=int helper(int); FULL_NAME=helper SIGNATURE=int(int) ORDER=1
   METHOD_PARAMETER_IN NAME= CODE=int TYPE_FULL_NAME=int ORDER=1
   METHOD_PARAMETER_OUT NAME= CODE=int TYPE_FULL_NAME=int ORDER=1
@@ -36,24 +30,39 @@
     METHOD_PARAMETER_OUT NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
     BLOCK CODE={ int y = out(global, F, M(x)); return global + y; } TYPE_FULL_NAME=void ORDER=2
       LOCAL NAME=global CODE=<global> global TYPE_FULL_NAME=int ORDER=0
-      LOCAL NAME=F CODE=<unknown> F TYPE_FULL_NAME=ANY ORDER=0
+      LOCAL NAME=UNDEF CODE=<unknown> UNDEF TYPE_FULL_NAME=ANY ORDER=0
       LOCAL NAME=y CODE=int y TYPE_FULL_NAME=int ORDER=1
       CALL NAME=<operator>.assignment CODE=y = out(global, F, M(x)) TYPE_FULL_NAME=void METHOD_FULL_NAME=<operator>.assignment ORDER=2 DISPATCH_TYPE=STATIC_DISPATCH
         IDENTIFIER NAME=y CODE=y TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
-        CALL NAME=out CODE=out(global, F, M(x)) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=out ORDER=2 ARGUMENT_INDEX=2 DISPATCH_TYPE=STATIC_DISPATCH
-          IDENTIFIER NAME=global CODE=<unknown> global TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
-          IDENTIFIER NAME=F CODE=<unknown> F TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
-          CALL NAME=M CODE=M(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=M ORDER=3 ARGUMENT_INDEX=3 DISPATCH_TYPE=STATIC_DISPATCH
-            IDENTIFIER NAME=x CODE=<unknown> x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+        CALL NAME=out CODE=out(global, F, M(x)) TYPE_FULL_NAME=int METHOD_FULL_NAME=out ORDER=2 ARGUMENT_INDEX=2 DISPATCH_TYPE=STATIC_DISPATCH
+          IDENTIFIER NAME=global CODE=<global> global TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+          CALL NAME=F CODE=F TYPE_FULL_NAME=char* METHOD_FULL_NAME=main.c:F:char*(0) SIGNATURE=char*(0) ORDER=2 ARGUMENT_INDEX=2 DISPATCH_TYPE=INLINED
+            BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
+              LITERAL CODE="%" UNDEF "d" TYPE_FULL_NAME=char* ORDER=1
+          CALL NAME=M CODE=M(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:M:ANY(1) SIGNATURE=ANY(1) ORDER=3 ARGUMENT_INDEX=3 DISPATCH_TYPE=INLINED
+            IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+            BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
+              CALL NAME=helper CODE=helper(x) TYPE_FULL_NAME=int METHOD_FULL_NAME=helper ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
+                IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
       RETURN CODE=return global + y; ORDER=3
         CALL NAME=<operator>.addition CODE=global + y TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.addition ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
           IDENTIFIER NAME=global CODE=<global> global TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
-          IDENTIFIER NAME=y CODE=<unknown> y TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
+          IDENTIFIER NAME=y CODE=y TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
     METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
   BLOCK TYPE_FULL_NAME=ANY ORDER=1
     LOCAL NAME=global CODE=int global TYPE_FULL_NAME=int ORDER=1
     METHOD_REF CODE=test TYPE_FULL_NAME=test METHOD_FULL_NAME=test ORDER=2
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
+
+METHOD NAME=F CODE=#define F "%" UNDEF "d" FULL_NAME=main.c:F:char*(0) SIGNATURE=char*(0) ORDER=1
+  BLOCK TYPE_FULL_NAME=ANY ORDER=1
+  METHOD_RETURN CODE=RET TYPE_FULL_NAME=char* ORDER=2
+
+METHOD NAME=M CODE=#define M(x) helper(x) FULL_NAME=main.c:M:ANY(1) SIGNATURE=ANY(1) ORDER=1
+  METHOD_PARAMETER_IN NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
+  METHOD_PARAMETER_OUT NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
+  BLOCK TYPE_FULL_NAME=ANY ORDER=2
+  METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=3
 
 METHOD NAME=out CODE=int out(int, const char *, int); FULL_NAME=out SIGNATURE=int(int,char*,int) ORDER=1
   METHOD_PARAMETER_IN NAME= CODE=int TYPE_FULL_NAME=int ORDER=1
@@ -70,19 +79,24 @@
   METHOD_PARAMETER_OUT NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
   BLOCK CODE={ int y = out(global, F, M(x)); return global + y; } TYPE_FULL_NAME=void ORDER=2
     LOCAL NAME=global CODE=<global> global TYPE_FULL_NAME=int ORDER=0
-    LOCAL NAME=F CODE=<unknown> F TYPE_FULL_NAME=ANY ORDER=0
+    LOCAL NAME=UNDEF CODE=<unknown> UNDEF TYPE_FULL_NAME=ANY ORDER=0
     LOCAL NAME=y CODE=int y TYPE_FULL_NAME=int ORDER=1
     CALL NAME=<operator>.assignment CODE=y = out(global, F, M(x)) TYPE_FULL_NAME=void METHOD_FULL_NAME=<operator>.assignment ORDER=2 DISPATCH_TYPE=STATIC_DISPATCH
       IDENTIFIER NAME=y CODE=y TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
-      CALL NAME=out CODE=out(global, F, M(x)) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=out ORDER=2 ARGUMENT_INDEX=2 DISPATCH_TYPE=STATIC_DISPATCH
-        IDENTIFIER NAME=global CODE=<unknown> global TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
-        IDENTIFIER NAME=F CODE=<unknown> F TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
-        CALL NAME=M CODE=M(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=M ORDER=3 ARGUMENT_INDEX=3 DISPATCH_TYPE=STATIC_DISPATCH
-          IDENTIFIER NAME=x CODE=<unknown> x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+      CALL NAME=out CODE=out(global, F, M(x)) TYPE_FULL_NAME=int METHOD_FULL_NAME=out ORDER=2 ARGUMENT_INDEX=2 DISPATCH_TYPE=STATIC_DISPATCH
+        IDENTIFIER NAME=global CODE=<global> global TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+        CALL NAME=F CODE=F TYPE_FULL_NAME=char* METHOD_FULL_NAME=main.c:F:char*(0) SIGNATURE=char*(0) ORDER=2 ARGUMENT_INDEX=2 DISPATCH_TYPE=INLINED
+          BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
+            LITERAL CODE="%" UNDEF "d" TYPE_FULL_NAME=char* ORDER=1
+        CALL NAME=M CODE=M(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:M:ANY(1) SIGNATURE=ANY(1) ORDER=3 ARGUMENT_INDEX=3 DISPATCH_TYPE=INLINED
+          IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
+          BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
+            CALL NAME=helper CODE=helper(x) TYPE_FULL_NAME=int METHOD_FULL_NAME=helper ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
+              IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
     RETURN CODE=return global + y; ORDER=3
       CALL NAME=<operator>.addition CODE=global + y TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.addition ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
         IDENTIFIER NAME=global CODE=<global> global TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=1
-        IDENTIFIER NAME=y CODE=<unknown> y TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
+        IDENTIFIER NAME=y CODE=y TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
 
 NODES|META_DATA LANGUAGE=NEWC
@@ -104,37 +118,46 @@
 NODES|TYPE NAME=int FULL_NAME=int TYPE_DECL_FULL_NAME=int
 NODES|TYPE NAME=test FULL_NAME=test TYPE_DECL_FULL_NAME=test
 NODES|TYPE NAME=void FULL_NAME=void TYPE_DECL_FULL_NAME=void
-EDGES|ARGUMENT test#12 -> test#13
 EDGES|ARGUMENT test#14 -> test#15
-EDGES|ARGUMENT test#15 -> test#16
-EDGES|ARGUMENT test#15 -> test#17
+EDGES|ARGUMENT test#17 -> test#18
+EDGES|ARGUMENT test#19 -> test#20
+EDGES|ARGUMENT test#20 -> test#21
+EDGES|ARGUMENT test#20 -> test#22
 EDGES|ARGUMENT test#7 -> test#8
 EDGES|ARGUMENT test#7 -> test#9
 EDGES|ARGUMENT test#9 -> test#10
 EDGES|ARGUMENT test#9 -> test#11
-EDGES|ARGUMENT test#9 -> test#12
-EDGES|CALL test#12 -> M#0
-EDGES|CALL test#15 -> <operator>.addition#0
+EDGES|ARGUMENT test#9 -> test#14
+EDGES|CALL test#11 -> main.c:F:char*(0)#0
+EDGES|CALL test#14 -> main.c:M:ANY(1)#0
+EDGES|CALL test#17 -> helper#0
+EDGES|CALL test#20 -> <operator>.addition#0
 EDGES|CALL test#7 -> <operator>.assignment#0
 EDGES|CALL test#9 -> out#0
 EDGES|CFG <includes>:<global>#0 -> <includes>:<global>#2
 EDGES|CFG <operator>.addition#0 -> <operator>.addition#5
 EDGES|CFG <operator>.assignment#0 -> <operator>.assignment#5
-EDGES|CFG M#0 -> M#4
 EDGES|CFG helper#0 -> helper#4
-EDGES|CFG main.c:<global>#0 -> main.c:<global>#22
+EDGES|CFG main.c:<global>#0 -> main.c:<global>#27
 EDGES|CFG main.c:<global>#1 -> test#8
-EDGES|CFG main.c:<global>#22 -> main.c:<global>#23
+EDGES|CFG main.c:<global>#27 -> main.c:<global>#28
+EDGES|CFG main.c:F:char*(0)#0 -> main.c:F:char*(0)#2
+EDGES|CFG main.c:M:ANY(1)#0 -> main.c:M:ANY(1)#4
 EDGES|CFG out#0 -> out#8
 EDGES|CFG test#10 -> test#11
 EDGES|CFG test#11 -> test#13
-EDGES|CFG test#12 -> test#9
-EDGES|CFG test#13 -> test#12
+EDGES|CFG test#11 -> test#15
+EDGES|CFG test#13 -> test#15
 EDGES|CFG test#14 -> test#18
+EDGES|CFG test#14 -> test#9
 EDGES|CFG test#15 -> test#14
-EDGES|CFG test#16 -> test#17
-EDGES|CFG test#17 -> test#15
-EDGES|CFG test#7 -> test#16
+EDGES|CFG test#17 -> test#9
+EDGES|CFG test#18 -> test#17
+EDGES|CFG test#19 -> test#23
+EDGES|CFG test#20 -> test#19
+EDGES|CFG test#21 -> test#22
+EDGES|CFG test#22 -> test#20
+EDGES|CFG test#7 -> test#21
 EDGES|CFG test#8 -> test#10
 EDGES|CFG test#9 -> test#7
 EDGES|CONTAINS <includes>:<global>#0 -> <includes>:<global>#1
@@ -143,6 +166,8 @@
 EDGES|CONTAINS D:main.c:<global> -> D:test
 EDGES|CONTAINS D:main.c:<global> -> helper#0
 EDGES|CONTAINS D:main.c:<global> -> main.c:<global>#0
+EDGES|CONTAINS D:main.c:<global> -> main.c:F:char*(0)#0
+EDGES|CONTAINS D:main.c:<global> -> main.c:M:ANY(1)#0
 EDGES|CONTAINS D:main.c:<global> -> out#0
 EDGES|CONTAINS F:<includes> -> <includes>:<global>#0
 EDGES|CONTAINS F:<includes> -> D:ANY
@@ -150,11 +175,10 @@
 EDGES|CONTAINS F:<includes> -> D:int
 EDGES|CONTAINS F:<includes> -> D:void
 EDGES|CONTAINS F:main.c -> D:main.c:<global>
-EDGES|CONTAINS M#0 -> M#2
 EDGES|CONTAINS helper#0 -> helper#3
 EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#1
-EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#20
-EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#22
+EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#25
+EDGES|CONTAINS main.c:<global>#0 -> main.c:<global>#27
 EDGES|CONTAINS main.c:<global>#1 -> test#10
 EDGES|CONTAINS main.c:<global>#1 -> test#11
 EDGES|CONTAINS main.c:<global>#1 -> test#12
@@ -163,10 +187,17 @@
 EDGES|CONTAINS main.c:<global>#1 -> test#15
 EDGES|CONTAINS main.c:<global>#1 -> test#16
 EDGES|CONTAINS main.c:<global>#1 -> test#17
+EDGES|CONTAINS main.c:<global>#1 -> test#18
+EDGES|CONTAINS main.c:<global>#1 -> test#19
+EDGES|CONTAINS main.c:<global>#1 -> test#20
+EDGES|CONTAINS main.c:<global>#1 -> test#21
+EDGES|CONTAINS main.c:<global>#1 -> test#22
 EDGES|CONTAINS main.c:<global>#1 -> test#3
 EDGES|CONTAINS main.c:<global>#1 -> test#7
 EDGES|CONTAINS main.c:<global>#1 -> test#8
 EDGES|CONTAINS main.c:<global>#1 -> test#9
+EDGES|CONTAINS main.c:F:char*(0)#0 -> main.c:F:char*(0)#1
+EDGES|CONTAINS main.c:M:ANY(1)#0 -> main.c:M:ANY(1)#3
 EDGES|CONTAINS out#0 -> out#7
 EDGES|EVAL_TYPE <includes>:<global>#1 -> T:ANY
 EDGES|EVAL_TYPE <includes>:<global>#2 -> T:ANY
@@ -182,18 +213,20 @@
 EDGES|EVAL_TYPE <operator>.assignment#4 -> T:ANY
 EDGES|EVAL_TYPE <operator>.assignment#5 -> T:ANY
 EDGES|EVAL_TYPE <operator>.assignment#6 -> T:ANY
-EDGES|EVAL_TYPE M#1 -> T:ANY
-EDGES|EVAL_TYPE M#2 -> T:ANY
-EDGES|EVAL_TYPE M#3 -> T:ANY
-EDGES|EVAL_TYPE M#4 -> T:ANY
 EDGES|EVAL_TYPE helper#1 -> T:int
 EDGES|EVAL_TYPE helper#2 -> T:int
 EDGES|EVAL_TYPE helper#3 -> T:ANY
 EDGES|EVAL_TYPE helper#4 -> T:int
-EDGES|EVAL_TYPE main.c:<global>#20 -> T:ANY
-EDGES|EVAL_TYPE main.c:<global>#21 -> T:int
-EDGES|EVAL_TYPE main.c:<global>#22 -> T:test
-EDGES|EVAL_TYPE main.c:<global>#23 -> T:ANY
+EDGES|EVAL_TYPE main.c:<global>#25 -> T:ANY
+EDGES|EVAL_TYPE main.c:<global>#26 -> T:int
+EDGES|EVAL_TYPE main.c:<global>#27 -> T:test
+EDGES|EVAL_TYPE main.c:<global>#28 -> T:ANY
+EDGES|EVAL_TYPE main.c:F:char*(0)#1 -> T:ANY
+EDGES|EVAL_TYPE main.c:F:char*(0)#2 -> T:char*
+EDGES|EVAL_TYPE main.c:M:ANY(1)#1 -> T:ANY
+EDGES|EVAL_TYPE main.c:M:ANY(1)#2 -> T:ANY
+EDGES|EVAL_TYPE main.c:M:ANY(1)#3 -> T:ANY
+EDGES|EVAL_TYPE main.c:M:ANY(1)#4 -> T:ANY
 EDGES|EVAL_TYPE out#1 -> T:int
 EDGES|EVAL_TYPE out#2 -> T:int
 EDGES|EVAL_TYPE out#3 -> T:char*
@@ -204,27 +237,32 @@
 EDGES|EVAL_TYPE out#8 -> T:int
 EDGES|EVAL_TYPE test#1 -> T:int
 EDGES|EVAL_TYPE test#10 -> T:int
-EDGES|EVAL_TYPE test#11 -> T:ANY
+EDGES|EVAL_TYPE test#11 -> T:char*
 EDGES|EVAL_TYPE test#12 -> T:ANY
-EDGES|EVAL_TYPE test#13 -> T:int
-EDGES|EVAL_TYPE test#15 -> T:ANY
-EDGES|EVAL_TYPE test#16 -> T:int
+EDGES|EVAL_TYPE test#13 -> T:char*
+EDGES|EVAL_TYPE test#14 -> T:ANY
+EDGES|EVAL_TYPE test#15 -> T:int
+EDGES|EVAL_TYPE test#16 -> T:ANY
 EDGES|EVAL_TYPE test#17 -> T:int
 EDGES|EVAL_TYPE test#18 -> T:int
 EDGES|EVAL_TYPE test#2 -> T:int
+EDGES|EVAL_TYPE test#20 -> T:ANY
+EDGES|EVAL_TYPE test#21 -> T:int
+EDGES|EVAL_TYPE test#22 -> T:int
+EDGES|EVAL_TYPE test#23 -> T:int
 EDGES|EVAL_TYPE test#3 -> T:void
 EDGES|EVAL_TYPE test#4 -> T:int
 EDGES|EVAL_TYPE test#5 -> T:ANY
 EDGES|EVAL_TYPE test#6 -> T:int
 EDGES|EVAL_TYPE test#7 -> T:void
 EDGES|EVAL_TYPE test#8 -> T:int
-EDGES|EVAL_TYPE test#9 -> T:ANY
+EDGES|EVAL_TYPE test#9 -> T:int
 EDGES|PARAMETER_LINK <operator>.addition#1 -> <operator>.addition#3
 EDGES|PARAMETER_LINK <operator>.addition#4 -> <operator>.addition#6
 EDGES|PARAMETER_LINK <operator>.assignment#1 -> <operator>.assignment#3
 EDGES|PARAMETER_LINK <operator>.assignment#4 -> <operator>.assignment#6
-EDGES|PARAMETER_LINK M#1 -> M#3
 EDGES|PARAMETER_LINK helper#1 -> helper#2
+EDGES|PARAMETER_LINK main.c:M:ANY(1)#1 -> main.c:M:ANY(1)#2
 EDGES|PARAMETER_LINK out#1 -> out#2
 EDGES|PARAMETER_LINK out#3 -> out#4
 EDGES|PARAMETER_LINK out#5 -> out#6
@@ -237,12 +275,11 @@
 EDGES|REF T:int -> D:int
 EDGES|REF T:test -> D:test
 EDGES|REF T:void -> D:void
-EDGES|REF main.c:<global>#22 -> main.c:<global>#1
+EDGES|REF main.c:<global>#27 -> main.c:<global>#1
 EDGES|REF test#10 -> test#4
-EDGES|REF test#11 -> test#5
-EDGES|REF test#13 -> test#1
-EDGES|REF test#16 -> test#4
-EDGES|REF test#17 -> test#6
+EDGES|REF test#18 -> test#1
+EDGES|REF test#21 -> test#4
+EDGES|REF test#22 -> test#6
 EDGES|REF test#8 -> test#6
 EDGES|SOURCE_FILE <includes>:<global>#0 -> F:<includes>
 EDGES|SOURCE_FILE D:ANY -> F:<includes>
@@ -257,42 +294,41 @@
 EDGES|SOURCE_FILE helper#0 -> F:main.c
 EDGES|SOURCE_FILE main.c:<global>#0 -> F:main.c
 EDGES|SOURCE_FILE main.c:<global>#1 -> F:main.c
+EDGES|SOURCE_FILE main.c:F:char*(0)#0 -> F:main.c
+EDGES|SOURCE_FILE main.c:M:ANY(1)#0 -> F:main.c
 EDGES|SOURCE_FILE out#0 -> F:main.c
-FLOWS|REACHING_DEF[<RET>] test#14 -> test#18
-FLOWS|REACHING_DEF[<global> global] test#16 -> test#15
-FLOWS|REACHING_DEF[<global> global] test#16 -> test#18
-FLOWS|REACHING_DEF[<unknown> F] test#11 -> test#10
-FLOWS|REACHING_DEF[<unknown> F] test#11 -> test#12
-FLOWS|REACHING_DEF[<unknown> F] test#11 -> test#18
-FLOWS|REACHING_DEF[<unknown> F] test#11 -> test#9
-FLOWS|REACHING_DEF[<unknown> global] test#10 -> test#11
-FLOWS|REACHING_DEF[<unknown> global] test#10 -> test#12
-FLOWS|REACHING_DEF[<unknown> global] test#10 -> test#16
-FLOWS|REACHING_DEF[<unknown> global] test#10 -> test#9
-FLOWS|REACHING_DEF[<unknown> x] test#13 -> test#12
-FLOWS|REACHING_DEF[<unknown> x] test#13 -> test#18
-FLOWS|REACHING_DEF[<unknown> x] test#13 -> test#2
-FLOWS|REACHING_DEF[<unknown> y] test#17 -> test#15
-FLOWS|REACHING_DEF[<unknown> y] test#17 -> test#18
-FLOWS|REACHING_DEF[M(x)] test#12 -> test#10
-FLOWS|REACHING_DEF[M(x)] test#12 -> test#11
-FLOWS|REACHING_DEF[M(x)] test#12 -> test#18
-FLOWS|REACHING_DEF[M(x)] test#12 -> test#9
+FLOWS|REACHING_DEF[<RET>] test#19 -> test#23
+FLOWS|REACHING_DEF[<global> global] test#10 -> test#11
+FLOWS|REACHING_DEF[<global> global] test#10 -> test#14
+FLOWS|REACHING_DEF[<global> global] test#10 -> test#21
+FLOWS|REACHING_DEF[<global> global] test#10 -> test#9
+FLOWS|REACHING_DEF[<global> global] test#21 -> test#20
+FLOWS|REACHING_DEF[<global> global] test#21 -> test#23
+FLOWS|REACHING_DEF[F] test#11 -> test#10
+FLOWS|REACHING_DEF[F] test#11 -> test#14
+FLOWS|REACHING_DEF[F] test#11 -> test#23
+FLOWS|REACHING_DEF[F] test#11 -> test#9
+FLOWS|REACHING_DEF[M(x)] test#14 -> test#10
+FLOWS|REACHING_DEF[M(x)] test#14 -> test#11
+FLOWS|REACHING_DEF[M(x)] test#14 -> test#23
+FLOWS|REACHING_DEF[M(x)] test#14 -> test#9
 FLOWS|REACHING_DEF[] <operator>.addition#0 -> <operator>.addition#1
 FLOWS|REACHING_DEF[] <operator>.addition#0 -> <operator>.addition#4
 FLOWS|REACHING_DEF[] <operator>.assignment#0 -> <operator>.assignment#1
 FLOWS|REACHING_DEF[] <operator>.assignment#0 -> <operator>.assignment#4
-FLOWS|REACHING_DEF[] M#0 -> M#1
 FLOWS|REACHING_DEF[] helper#0 -> helper#1
 FLOWS|REACHING_DEF[] helper#1 -> helper#2
 FLOWS|REACHING_DEF[] helper#1 -> helper#4
-FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#22
+FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#27
 FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#1
 FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#10
 FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#11
 FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#13
-FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#16
-FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#17
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#15
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#18
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#21
+FLOWS|REACHING_DEF[] main.c:<global>#1 -> test#22
+FLOWS|REACHING_DEF[] main.c:M:ANY(1)#0 -> main.c:M:ANY(1)#1
 FLOWS|REACHING_DEF[] out#0 -> out#1
 FLOWS|REACHING_DEF[] out#0 -> out#3
 FLOWS|REACHING_DEF[] out#0 -> out#5
@@ -302,23 +338,33 @@
 FLOWS|REACHING_DEF[] out#3 -> out#8
 FLOWS|REACHING_DEF[] out#5 -> out#6
 FLOWS|REACHING_DEF[] out#5 -> out#8
-FLOWS|REACHING_DEF[global + y] test#15 -> test#14
-FLOWS|REACHING_DEF[global + y] test#15 -> test#18
-FLOWS|REACHING_DEF[out(global, F, M(x))] test#9 -> test#18
+FLOWS|REACHING_DEF[global + y] test#20 -> test#19
+FLOWS|REACHING_DEF[global + y] test#20 -> test#23
+FLOWS|REACHING_DEF[helper(x)] test#17 -> test#23
+FLOWS|REACHING_DEF[out(global, F, M(x))] test#9 -> test#23
 FLOWS|REACHING_DEF[out(global, F, M(x))] test#9 -> test#7
 FLOWS|REACHING_DEF[out(global, F, M(x))] test#9 -> test#8
 FLOWS|REACHING_DEF[p1] <operator>.addition#1 -> <operator>.addition#3
 FLOWS|REACHING_DEF[p1] <operator>.addition#1 -> <operator>.addition#5
 FLOWS|REACHING_DEF[p1] <operator>.assignment#1 -> <operator>.assignment#3
 FLOWS|REACHING_DEF[p1] <operator>.assignment#1 -> <operator>.assignment#5
-FLOWS|REACHING_DEF[p1] M#1 -> M#3
-FLOWS|REACHING_DEF[p1] M#1 -> M#4
+FLOWS|REACHING_DEF[p1] main.c:M:ANY(1)#1 -> main.c:M:ANY(1)#2
+FLOWS|REACHING_DEF[p1] main.c:M:ANY(1)#1 -> main.c:M:ANY(1)#4
 FLOWS|REACHING_DEF[p2] <operator>.addition#4 -> <operator>.addition#5
 FLOWS|REACHING_DEF[p2] <operator>.addition#4 -> <operator>.addition#6
 FLOWS|REACHING_DEF[p2] <operator>.assignment#4 -> <operator>.assignment#5
 FLOWS|REACHING_DEF[p2] <operator>.assignment#4 -> <operator>.assignment#6
-FLOWS|REACHING_DEF[x] test#1 -> test#13
+FLOWS|REACHING_DEF[x] test#1 -> test#15
 FLOWS|REACHING_DEF[x] test#1 -> test#2
-FLOWS|REACHING_DEF[y = out(global, F, M(x))] test#7 -> test#18
-FLOWS|REACHING_DEF[y] test#8 -> test#17
+FLOWS|REACHING_DEF[x] test#15 -> test#14
+FLOWS|REACHING_DEF[x] test#15 -> test#18
+FLOWS|REACHING_DEF[x] test#15 -> test#2
+FLOWS|REACHING_DEF[x] test#15 -> test#23
+FLOWS|REACHING_DEF[x] test#18 -> test#17
+FLOWS|REACHING_DEF[x] test#18 -> test#2
+FLOWS|REACHING_DEF[x] test#18 -> test#23
+FLOWS|REACHING_DEF[y = out(global, F, M(x))] test#7 -> test#23
+FLOWS|REACHING_DEF[y] test#22 -> test#20
+FLOWS|REACHING_DEF[y] test#22 -> test#23
+FLOWS|REACHING_DEF[y] test#8 -> test#22
 FLOWS|REACHING_DEF[y] test#8 -> test#7
