--- joern
+++ before
@@ -78,8 +78,6 @@
 EDGES|CALL invoke#8 -> <operator>.addition#0
 EDGES|CFG <includes>:<global>#0 -> <includes>:<global>#2
 EDGES|CFG <operator>.addition#0 -> <operator>.addition#5
-EDGES|CFG a value=name.h:<global>#0 -> a value=name.h:<global>#2
-EDGES|CFG a value=name.h:VALUE:ANY(1)#0 -> a value=name.h:VALUE:ANY(1)#4
 EDGES|CFG invoke#0 -> invoke#6
 EDGES|CFG invoke#10 -> invoke#8
 EDGES|CFG invoke#4 -> invoke#11
@@ -178,7 +176,6 @@
 FLOWS|REACHING_DEF[VALUE(x)] invoke#5 -> invoke#4
 FLOWS|REACHING_DEF[] <operator>.addition#0 -> <operator>.addition#1
 FLOWS|REACHING_DEF[] <operator>.addition#0 -> <operator>.addition#4
-FLOWS|REACHING_DEF[] a value=name.h:VALUE:ANY(1)#0 -> a value=name.h:VALUE:ANY(1)#1
 FLOWS|REACHING_DEF[] invoke#0 -> invoke#1
 FLOWS|REACHING_DEF[] invoke#0 -> invoke#10
 FLOWS|REACHING_DEF[] invoke#0 -> invoke#6
