--- joern
+++ baseline
@@ -2,26 +2,24 @@
   BLOCK TYPE_FULL_NAME=ANY ORDER=1
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
 
-METHOD NAME=<operator>.cast FULL_NAME=<operator>.cast ORDER=0
+METHOD NAME=<operator>.pointerCall FULL_NAME=<operator>.pointerCall ORDER=0
   METHOD_PARAMETER_IN NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
   BLOCK TYPE_FULL_NAME=ANY ORDER=1 ARGUMENT_INDEX=1
   METHOD_PARAMETER_OUT NAME=p1 CODE=p1 TYPE_FULL_NAME=ANY ORDER=1
-  METHOD_PARAMETER_IN NAME=p2 CODE=p2 TYPE_FULL_NAME=ANY ORDER=2
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=ANY ORDER=2
-  METHOD_PARAMETER_OUT NAME=p2 CODE=p2 TYPE_FULL_NAME=ANY ORDER=2
 
 METHOD NAME=f CODE=int f(int x) { return C(T,x); } FULL_NAME=f SIGNATURE=int(int) ORDER=1
   METHOD_PARAMETER_IN NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
   METHOD_PARAMETER_OUT NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
   BLOCK CODE={ return C(T,x); } TYPE_FULL_NAME=void ORDER=2
+    LOCAL NAME=T CODE=<unknown> T TYPE_FULL_NAME=ANY ORDER=0
     RETURN CODE=return C(T,x); ORDER=1
       CALL NAME=C CODE=C(T,x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:C:ANY(2) SIGNATURE=ANY(2) ORDER=1 DISPATCH_TYPE=INLINED
-        TYPE_REF CODE=T TYPE_FULL_NAME=T ORDER=1 ARGUMENT_INDEX=1
-        IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
-        BLOCK TYPE_FULL_NAME=ANY ORDER=3 ARGUMENT_INDEX=3
-          CALL NAME=<operator>.cast CODE=(T)(x) TYPE_FULL_NAME=T METHOD_FULL_NAME=<operator>.cast ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
-            TYPE_REF CODE=T TYPE_FULL_NAME=T ORDER=1 ARGUMENT_INDEX=1
-            IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
+        IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=2
+        BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
+          CALL NAME=<operator>.pointerCall CODE=(  T)(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.pointerCall ORDER=1 DISPATCH_TYPE=DYNAMIC_DISPATCH
+            IDENTIFIER NAME=T CODE=<unknown> T TYPE_FULL_NAME=ANY ORDER=1
+            IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=1
   METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
 
 METHOD NAME=<global> CODE=<global> FULL_NAME=main.c:<global> ORDER=1
@@ -29,14 +27,14 @@
     METHOD_PARAMETER_IN NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
     METHOD_PARAMETER_OUT NAME=x CODE=int x TYPE_FULL_NAME=int ORDER=1
     BLOCK CODE={ return C(T,x); } TYPE_FULL_NAME=void ORDER=2
+      LOCAL NAME=T CODE=<unknown> T TYPE_FULL_NAME=ANY ORDER=0
       RETURN CODE=return C(T,x); ORDER=1
         CALL NAME=C CODE=C(T,x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=main.c:C:ANY(2) SIGNATURE=ANY(2) ORDER=1 DISPATCH_TYPE=INLINED
-          TYPE_REF CODE=T TYPE_FULL_NAME=T ORDER=1 ARGUMENT_INDEX=1
-          IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
-          BLOCK TYPE_FULL_NAME=ANY ORDER=3 ARGUMENT_INDEX=3
-            CALL NAME=<operator>.cast CODE=(T)(x) TYPE_FULL_NAME=T METHOD_FULL_NAME=<operator>.cast ORDER=1 DISPATCH_TYPE=STATIC_DISPATCH
-              TYPE_REF CODE=T TYPE_FULL_NAME=T ORDER=1 ARGUMENT_INDEX=1
-              IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=2
+          IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=1 ARGUMENT_INDEX=2
+          BLOCK TYPE_FULL_NAME=ANY ORDER=2 ARGUMENT_INDEX=2
+            CALL NAME=<operator>.pointerCall CODE=(  T)(x) TYPE_FULL_NAME=ANY METHOD_FULL_NAME=<operator>.pointerCall ORDER=1 DISPATCH_TYPE=DYNAMIC_DISPATCH
+              IDENTIFIER NAME=T CODE=<unknown> T TYPE_FULL_NAME=ANY ORDER=1
+              IDENTIFIER NAME=x CODE=x TYPE_FULL_NAME=int ORDER=2 ARGUMENT_INDEX=1
     METHOD_RETURN CODE=RET TYPE_FULL_NAME=int ORDER=3
   BLOCK TYPE_FULL_NAME=ANY ORDER=1
     TYPE_DECL NAME=T CODE=typedef int T; FULL_NAME=T ORDER=1
@@ -70,29 +68,25 @@
 NODES|TYPE NAME=f FULL_NAME=f TYPE_DECL_FULL_NAME=f
 NODES|TYPE NAME=int FULL_NAME=int TYPE_DECL_FULL_NAME=int
 NODES|TYPE NAME=void FULL_NAME=void TYPE_DECL_FULL_NAME=void
-EDGES|ARGUMENT f#4 -> f#5
 EDGES|ARGUMENT f#5 -> f#6
-EDGES|ARGUMENT f#5 -> f#7
-EDGES|ARGUMENT f#9 -> f#10
+EDGES|ARGUMENT f#6 -> f#7
 EDGES|ARGUMENT f#9 -> f#11
-EDGES|CALL f#5 -> main.c:C:ANY(2)#0
-EDGES|CALL f#9 -> <operator>.cast#0
+EDGES|CALL f#6 -> main.c:C:ANY(2)#0
 EDGES|CFG <includes>:<global>#0 -> <includes>:<global>#2
-EDGES|CFG <operator>.cast#0 -> <operator>.cast#5
-EDGES|CFG f#0 -> f#6
+EDGES|CFG <operator>.pointerCall#0 -> <operator>.pointerCall#4
+EDGES|CFG f#0 -> f#7
 EDGES|CFG f#10 -> f#11
 EDGES|CFG f#11 -> f#9
-EDGES|CFG f#4 -> f#12
-EDGES|CFG f#5 -> f#10
-EDGES|CFG f#5 -> f#4
-EDGES|CFG f#6 -> f#7
-EDGES|CFG f#7 -> f#5
-EDGES|CFG f#9 -> f#4
+EDGES|CFG f#5 -> f#12
+EDGES|CFG f#6 -> f#10
+EDGES|CFG f#6 -> f#5
+EDGES|CFG f#7 -> f#6
+EDGES|CFG f#9 -> f#5
 EDGES|CFG main.c:<global>#0 -> main.c:<global>#16
 EDGES|CFG main.c:<global>#16 -> main.c:<global>#17
 EDGES|CFG main.c:C:ANY(2)#0 -> main.c:C:ANY(2)#6
 EDGES|CONTAINS <includes>:<global>#0 -> <includes>:<global>#1
-EDGES|CONTAINS <operator>.cast#0 -> <operator>.cast#2
+EDGES|CONTAINS <operator>.pointerCall#0 -> <operator>.pointerCall#2
 EDGES|CONTAINS D:main.c:<global> -> D:f
 EDGES|CONTAINS D:main.c:<global> -> main.c:<global>#0
 EDGES|CONTAINS D:main.c:<global> -> main.c:C:ANY(2)#0
@@ -104,7 +98,6 @@
 EDGES|CONTAINS f#0 -> f#10
 EDGES|CONTAINS f#0 -> f#11
 EDGES|CONTAINS f#0 -> f#3
-EDGES|CONTAINS f#0 -> f#4
 EDGES|CONTAINS f#0 -> f#5
 EDGES|CONTAINS f#0 -> f#6
 EDGES|CONTAINS f#0 -> f#7
@@ -117,23 +110,21 @@
 EDGES|CONTAINS main.c:C:ANY(2)#0 -> main.c:C:ANY(2)#5
 EDGES|EVAL_TYPE <includes>:<global>#1 -> T:ANY
 EDGES|EVAL_TYPE <includes>:<global>#2 -> T:ANY
-EDGES|EVAL_TYPE <operator>.cast#1 -> T:ANY
-EDGES|EVAL_TYPE <operator>.cast#2 -> T:ANY
-EDGES|EVAL_TYPE <operator>.cast#3 -> T:ANY
-EDGES|EVAL_TYPE <operator>.cast#4 -> T:ANY
-EDGES|EVAL_TYPE <operator>.cast#5 -> T:ANY
-EDGES|EVAL_TYPE <operator>.cast#6 -> T:ANY
+EDGES|EVAL_TYPE <operator>.pointerCall#1 -> T:ANY
+EDGES|EVAL_TYPE <operator>.pointerCall#2 -> T:ANY
+EDGES|EVAL_TYPE <operator>.pointerCall#3 -> T:ANY
+EDGES|EVAL_TYPE <operator>.pointerCall#4 -> T:ANY
 EDGES|EVAL_TYPE f#1 -> T:int
-EDGES|EVAL_TYPE f#10 -> T:T
+EDGES|EVAL_TYPE f#10 -> T:ANY
 EDGES|EVAL_TYPE f#11 -> T:int
 EDGES|EVAL_TYPE f#12 -> T:int
 EDGES|EVAL_TYPE f#2 -> T:int
 EDGES|EVAL_TYPE f#3 -> T:void
-EDGES|EVAL_TYPE f#5 -> T:ANY
-EDGES|EVAL_TYPE f#6 -> T:T
+EDGES|EVAL_TYPE f#4 -> T:ANY
+EDGES|EVAL_TYPE f#6 -> T:ANY
 EDGES|EVAL_TYPE f#7 -> T:int
 EDGES|EVAL_TYPE f#8 -> T:ANY
-EDGES|EVAL_TYPE f#9 -> T:T
+EDGES|EVAL_TYPE f#9 -> T:ANY
 EDGES|EVAL_TYPE main.c:<global>#14 -> T:ANY
 EDGES|EVAL_TYPE main.c:<global>#16 -> T:f
 EDGES|EVAL_TYPE main.c:<global>#17 -> T:ANY
@@ -143,8 +134,7 @@
 EDGES|EVAL_TYPE main.c:C:ANY(2)#4 -> T:ANY
 EDGES|EVAL_TYPE main.c:C:ANY(2)#5 -> T:ANY
 EDGES|EVAL_TYPE main.c:C:ANY(2)#6 -> T:ANY
-EDGES|PARAMETER_LINK <operator>.cast#1 -> <operator>.cast#3
-EDGES|PARAMETER_LINK <operator>.cast#4 -> <operator>.cast#6
+EDGES|PARAMETER_LINK <operator>.pointerCall#1 -> <operator>.pointerCall#3
 EDGES|PARAMETER_LINK f#1 -> f#2
 EDGES|PARAMETER_LINK main.c:C:ANY(2)#1 -> main.c:C:ANY(2)#2
 EDGES|PARAMETER_LINK main.c:C:ANY(2)#3 -> main.c:C:ANY(2)#4
@@ -156,6 +146,7 @@
 EDGES|REF T:f -> D:f
 EDGES|REF T:int -> D:int
 EDGES|REF T:void -> D:void
+EDGES|REF f#10 -> f#4
 EDGES|REF f#11 -> f#1
 EDGES|REF main.c:<global>#16 -> f#0
 EDGES|SOURCE_FILE <includes>:<global>#0 -> F:<includes>
@@ -171,29 +162,22 @@
 EDGES|SOURCE_FILE main.c:<global>#0 -> F:main.c
 EDGES|SOURCE_FILE main.c:<global>#15 -> F:main.c
 EDGES|SOURCE_FILE main.c:C:ANY(2)#0 -> F:main.c
-FLOWS|REACHING_DEF[(T)(x)] f#9 -> f#12
-FLOWS|REACHING_DEF[<RET>] f#4 -> f#12
-FLOWS|REACHING_DEF[C(T,x)] f#5 -> f#12
-FLOWS|REACHING_DEF[C(T,x)] f#5 -> f#4
-FLOWS|REACHING_DEF[T] f#10 -> f#9
-FLOWS|REACHING_DEF[T] f#6 -> f#5
-FLOWS|REACHING_DEF[T] f#6 -> f#7
-FLOWS|REACHING_DEF[] <operator>.cast#0 -> <operator>.cast#1
-FLOWS|REACHING_DEF[] <operator>.cast#0 -> <operator>.cast#4
+FLOWS|REACHING_DEF[(  T)(x)] f#9 -> f#12
+FLOWS|REACHING_DEF[<RET>] f#5 -> f#12
+FLOWS|REACHING_DEF[C(T,x)] f#6 -> f#12
+FLOWS|REACHING_DEF[C(T,x)] f#6 -> f#5
+FLOWS|REACHING_DEF[] <operator>.pointerCall#0 -> <operator>.pointerCall#1
 FLOWS|REACHING_DEF[] f#0 -> f#1
 FLOWS|REACHING_DEF[] f#0 -> f#10
 FLOWS|REACHING_DEF[] f#0 -> f#11
-FLOWS|REACHING_DEF[] f#0 -> f#6
 FLOWS|REACHING_DEF[] f#0 -> f#7
 FLOWS|REACHING_DEF[] main.c:<global>#0 -> main.c:<global>#16
 FLOWS|REACHING_DEF[] main.c:C:ANY(2)#0 -> main.c:C:ANY(2)#1
 FLOWS|REACHING_DEF[] main.c:C:ANY(2)#0 -> main.c:C:ANY(2)#3
-FLOWS|REACHING_DEF[p1] <operator>.cast#1 -> <operator>.cast#3
-FLOWS|REACHING_DEF[p1] <operator>.cast#1 -> <operator>.cast#5
+FLOWS|REACHING_DEF[p1] <operator>.pointerCall#1 -> <operator>.pointerCall#3
+FLOWS|REACHING_DEF[p1] <operator>.pointerCall#1 -> <operator>.pointerCall#4
 FLOWS|REACHING_DEF[p1] main.c:C:ANY(2)#1 -> main.c:C:ANY(2)#2
 FLOWS|REACHING_DEF[p1] main.c:C:ANY(2)#1 -> main.c:C:ANY(2)#6
-FLOWS|REACHING_DEF[p2] <operator>.cast#4 -> <operator>.cast#5
-FLOWS|REACHING_DEF[p2] <operator>.cast#4 -> <operator>.cast#6
 FLOWS|REACHING_DEF[p2] main.c:C:ANY(2)#3 -> main.c:C:ANY(2)#4
 FLOWS|REACHING_DEF[p2] main.c:C:ANY(2)#3 -> main.c:C:ANY(2)#6
 FLOWS|REACHING_DEF[x] f#1 -> f#2
@@ -204,4 +188,4 @@
 FLOWS|REACHING_DEF[x] f#7 -> f#11
 FLOWS|REACHING_DEF[x] f#7 -> f#12
 FLOWS|REACHING_DEF[x] f#7 -> f#2
-FLOWS|REACHING_DEF[x] f#7 -> f#5
+FLOWS|REACHING_DEF[x] f#7 -> f#6
